// @aws-sdk/client-connect/dist-cjs/models/models_1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HierarchyStructureUpdate = exports.HierarchyLevelUpdate = exports.UpdateUserHierarchyGroupNameRequest = exports.UpdateUserHierarchyRequest = exports.UpdateSecurityProfileRequest = exports.UpdateRoutingProfileQueuesRequest = exports.UpdateRoutingProfileNameRequest = exports.UpdateRoutingProfileDefaultOutboundQueueRequest = exports.UpdateRoutingProfileConcurrencyRequest = exports.UpdateQuickConnectNameRequest = exports.UpdateQuickConnectConfigRequest = exports.UpdateQueueStatusRequest = exports.UpdateQueueOutboundCallerConfigRequest = exports.UpdateQueueNameRequest = exports.UpdateQueueMaxContactsRequest = exports.UpdateQueueHoursOfOperationRequest = exports.UpdateInstanceStorageConfigRequest = exports.UpdateInstanceAttributeRequest = exports.UpdateHoursOfOperationRequest = exports.UpdateContactScheduleResponse = exports.UpdateContactScheduleRequest = exports.UpdateContactFlowNameRequest = exports.UpdateContactFlowModuleMetadataResponse = exports.UpdateContactFlowModuleMetadataRequest = exports.UpdateContactFlowModuleContentResponse = exports.UpdateContactFlowModuleContentRequest = exports.UpdateContactFlowMetadataRequest = exports.UpdateContactFlowContentRequest = exports.UpdateContactAttributesResponse = exports.UpdateContactAttributesRequest = exports.UpdateContactResponse = exports.UpdateContactRequest = exports.UpdateAgentStatusRequest = exports.UntagResourceRequest = exports.TagResourceRequest = exports.SuspendContactRecordingResponse = exports.SuspendContactRecordingRequest = exports.StopContactStreamingResponse = exports.StopContactStreamingRequest = exports.StopContactRecordingResponse = exports.StopContactRecordingRequest = exports.StopContactResponse = exports.StopContactRequest = exports.StartTaskContactResponse = exports.StartTaskContactRequest = exports.Reference = exports.StartOutboundVoiceContactResponse = exports.StartOutboundVoiceContactRequest = exports.TrafficType = exports.AnswerMachineDetectionConfig = void 0;
exports.UpdateUserSecurityProfilesRequest = exports.UpdateUserRoutingProfileRequest = exports.UpdateUserPhoneConfigRequest = exports.UpdateUserIdentityInfoRequest = exports.UpdateUserHierarchyStructureRequest = void 0;
var AnswerMachineDetectionConfig;
(function (AnswerMachineDetectionConfig) {
    AnswerMachineDetectionConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AnswerMachineDetectionConfig = exports.AnswerMachineDetectionConfig || (exports.AnswerMachineDetectionConfig = {}));
var TrafficType;
(function (TrafficType) {
    TrafficType["CAMPAIGN"] = "CAMPAIGN";
    TrafficType["GENERAL"] = "GENERAL";
})(TrafficType = exports.TrafficType || (exports.TrafficType = {}));
var StartOutboundVoiceContactRequest;
(function (StartOutboundVoiceContactRequest) {
    StartOutboundVoiceContactRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StartOutboundVoiceContactRequest = exports.StartOutboundVoiceContactRequest || (exports.StartOutboundVoiceContactRequest = {}));
var StartOutboundVoiceContactResponse;
(function (StartOutboundVoiceContactResponse) {
    StartOutboundVoiceContactResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StartOutboundVoiceContactResponse = exports.StartOutboundVoiceContactResponse || (exports.StartOutboundVoiceContactResponse = {}));
var Reference;
(function (Reference) {
    Reference.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Reference = exports.Reference || (exports.Reference = {}));
var StartTaskContactRequest;
(function (StartTaskContactRequest) {
    StartTaskContactRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StartTaskContactRequest = exports.StartTaskContactRequest || (exports.StartTaskContactRequest = {}));
var StartTaskContactResponse;
(function (StartTaskContactResponse) {
    StartTaskContactResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StartTaskContactResponse = exports.StartTaskContactResponse || (exports.StartTaskContactResponse = {}));
var StopContactRequest;
(function (StopContactRequest) {
    StopContactRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StopContactRequest = exports.StopContactRequest || (exports.StopContactRequest = {}));
var StopContactResponse;
(function (StopContactResponse) {
    StopContactResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StopContactResponse = exports.StopContactResponse || (exports.StopContactResponse = {}));
var StopContactRecordingRequest;
(function (StopContactRecordingRequest) {
    StopContactRecordingRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StopContactRecordingRequest = exports.StopContactRecordingRequest || (exports.StopContactRecordingRequest = {}));
var StopContactRecordingResponse;
(function (StopContactRecordingResponse) {
    StopContactRecordingResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StopContactRecordingResponse = exports.StopContactRecordingResponse || (exports.StopContactRecordingResponse = {}));
var StopContactStreamingRequest;
(function (StopContactStreamingRequest) {
    StopContactStreamingRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StopContactStreamingRequest = exports.StopContactStreamingRequest || (exports.StopContactStreamingRequest = {}));
var StopContactStreamingResponse;
(function (StopContactStreamingResponse) {
    StopContactStreamingResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StopContactStreamingResponse = exports.StopContactStreamingResponse || (exports.StopContactStreamingResponse = {}));
var SuspendContactRecordingRequest;
(function (SuspendContactRecordingRequest) {
    SuspendContactRecordingRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SuspendContactRecordingRequest = exports.SuspendContactRecordingRequest || (exports.SuspendContactRecordingRequest = {}));
var SuspendContactRecordingResponse;
(function (SuspendContactRecordingResponse) {
    SuspendContactRecordingResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SuspendContactRecordingResponse = exports.SuspendContactRecordingResponse || (exports.SuspendContactRecordingResponse = {}));
var TagResourceRequest;
(function (TagResourceRequest) {
    TagResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagResourceRequest = exports.TagResourceRequest || (exports.TagResourceRequest = {}));
var UntagResourceRequest;
(function (UntagResourceRequest) {
    UntagResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UntagResourceRequest = exports.UntagResourceRequest || (exports.UntagResourceRequest = {}));
var UpdateAgentStatusRequest;
(function (UpdateAgentStatusRequest) {
    UpdateAgentStatusRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateAgentStatusRequest = exports.UpdateAgentStatusRequest || (exports.UpdateAgentStatusRequest = {}));
var UpdateContactRequest;
(function (UpdateContactRequest) {
    UpdateContactRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateContactRequest = exports.UpdateContactRequest || (exports.UpdateContactRequest = {}));
var UpdateContactResponse;
(function (UpdateContactResponse) {
    UpdateContactResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateContactResponse = exports.UpdateContactResponse || (exports.UpdateContactResponse = {}));
var UpdateContactAttributesRequest;
(function (UpdateContactAttributesRequest) {
    UpdateContactAttributesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateContactAttributesRequest = exports.UpdateContactAttributesRequest || (exports.UpdateContactAttributesRequest = {}));
var UpdateContactAttributesResponse;
(function (UpdateContactAttributesResponse) {
    UpdateContactAttributesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateContactAttributesResponse = exports.UpdateContactAttributesResponse || (exports.UpdateContactAttributesResponse = {}));
var UpdateContactFlowContentRequest;
(function (UpdateContactFlowContentRequest) {
    UpdateContactFlowContentRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateContactFlowContentRequest = exports.UpdateContactFlowContentRequest || (exports.UpdateContactFlowContentRequest = {}));
var UpdateContactFlowMetadataRequest;
(function (UpdateContactFlowMetadataRequest) {
    UpdateContactFlowMetadataRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateContactFlowMetadataRequest = exports.UpdateContactFlowMetadataRequest || (exports.UpdateContactFlowMetadataRequest = {}));
var UpdateContactFlowModuleContentRequest;
(function (UpdateContactFlowModuleContentRequest) {
    UpdateContactFlowModuleContentRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateContactFlowModuleContentRequest = exports.UpdateContactFlowModuleContentRequest || (exports.UpdateContactFlowModuleContentRequest = {}));
var UpdateContactFlowModuleContentResponse;
(function (UpdateContactFlowModuleContentResponse) {
    UpdateContactFlowModuleContentResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateContactFlowModuleContentResponse = exports.UpdateContactFlowModuleContentResponse || (exports.UpdateContactFlowModuleContentResponse = {}));
var UpdateContactFlowModuleMetadataRequest;
(function (UpdateContactFlowModuleMetadataRequest) {
    UpdateContactFlowModuleMetadataRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateContactFlowModuleMetadataRequest = exports.UpdateContactFlowModuleMetadataRequest || (exports.UpdateContactFlowModuleMetadataRequest = {}));
var UpdateContactFlowModuleMetadataResponse;
(function (UpdateContactFlowModuleMetadataResponse) {
    UpdateContactFlowModuleMetadataResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateContactFlowModuleMetadataResponse = exports.UpdateContactFlowModuleMetadataResponse || (exports.UpdateContactFlowModuleMetadataResponse = {}));
var UpdateContactFlowNameRequest;
(function (UpdateContactFlowNameRequest) {
    UpdateContactFlowNameRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateContactFlowNameRequest = exports.UpdateContactFlowNameRequest || (exports.UpdateContactFlowNameRequest = {}));
var UpdateContactScheduleRequest;
(function (UpdateContactScheduleRequest) {
    UpdateContactScheduleRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateContactScheduleRequest = exports.UpdateContactScheduleRequest || (exports.UpdateContactScheduleRequest = {}));
var UpdateContactScheduleResponse;
(function (UpdateContactScheduleResponse) {
    UpdateContactScheduleResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateContactScheduleResponse = exports.UpdateContactScheduleResponse || (exports.UpdateContactScheduleResponse = {}));
var UpdateHoursOfOperationRequest;
(function (UpdateHoursOfOperationRequest) {
    UpdateHoursOfOperationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateHoursOfOperationRequest = exports.UpdateHoursOfOperationRequest || (exports.UpdateHoursOfOperationRequest = {}));
var UpdateInstanceAttributeRequest;
(function (UpdateInstanceAttributeRequest) {
    UpdateInstanceAttributeRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateInstanceAttributeRequest = exports.UpdateInstanceAttributeRequest || (exports.UpdateInstanceAttributeRequest = {}));
var UpdateInstanceStorageConfigRequest;
(function (UpdateInstanceStorageConfigRequest) {
    UpdateInstanceStorageConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateInstanceStorageConfigRequest = exports.UpdateInstanceStorageConfigRequest || (exports.UpdateInstanceStorageConfigRequest = {}));
var UpdateQueueHoursOfOperationRequest;
(function (UpdateQueueHoursOfOperationRequest) {
    UpdateQueueHoursOfOperationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateQueueHoursOfOperationRequest = exports.UpdateQueueHoursOfOperationRequest || (exports.UpdateQueueHoursOfOperationRequest = {}));
var UpdateQueueMaxContactsRequest;
(function (UpdateQueueMaxContactsRequest) {
    UpdateQueueMaxContactsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateQueueMaxContactsRequest = exports.UpdateQueueMaxContactsRequest || (exports.UpdateQueueMaxContactsRequest = {}));
var UpdateQueueNameRequest;
(function (UpdateQueueNameRequest) {
    UpdateQueueNameRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateQueueNameRequest = exports.UpdateQueueNameRequest || (exports.UpdateQueueNameRequest = {}));
var UpdateQueueOutboundCallerConfigRequest;
(function (UpdateQueueOutboundCallerConfigRequest) {
    UpdateQueueOutboundCallerConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateQueueOutboundCallerConfigRequest = exports.UpdateQueueOutboundCallerConfigRequest || (exports.UpdateQueueOutboundCallerConfigRequest = {}));
var UpdateQueueStatusRequest;
(function (UpdateQueueStatusRequest) {
    UpdateQueueStatusRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateQueueStatusRequest = exports.UpdateQueueStatusRequest || (exports.UpdateQueueStatusRequest = {}));
var UpdateQuickConnectConfigRequest;
(function (UpdateQuickConnectConfigRequest) {
    UpdateQuickConnectConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateQuickConnectConfigRequest = exports.UpdateQuickConnectConfigRequest || (exports.UpdateQuickConnectConfigRequest = {}));
var UpdateQuickConnectNameRequest;
(function (UpdateQuickConnectNameRequest) {
    UpdateQuickConnectNameRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateQuickConnectNameRequest = exports.UpdateQuickConnectNameRequest || (exports.UpdateQuickConnectNameRequest = {}));
var UpdateRoutingProfileConcurrencyRequest;
(function (UpdateRoutingProfileConcurrencyRequest) {
    UpdateRoutingProfileConcurrencyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateRoutingProfileConcurrencyRequest = exports.UpdateRoutingProfileConcurrencyRequest || (exports.UpdateRoutingProfileConcurrencyRequest = {}));
var UpdateRoutingProfileDefaultOutboundQueueRequest;
(function (UpdateRoutingProfileDefaultOutboundQueueRequest) {
    UpdateRoutingProfileDefaultOutboundQueueRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateRoutingProfileDefaultOutboundQueueRequest = exports.UpdateRoutingProfileDefaultOutboundQueueRequest || (exports.UpdateRoutingProfileDefaultOutboundQueueRequest = {}));
var UpdateRoutingProfileNameRequest;
(function (UpdateRoutingProfileNameRequest) {
    UpdateRoutingProfileNameRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateRoutingProfileNameRequest = exports.UpdateRoutingProfileNameRequest || (exports.UpdateRoutingProfileNameRequest = {}));
var UpdateRoutingProfileQueuesRequest;
(function (UpdateRoutingProfileQueuesRequest) {
    UpdateRoutingProfileQueuesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateRoutingProfileQueuesRequest = exports.UpdateRoutingProfileQueuesRequest || (exports.UpdateRoutingProfileQueuesRequest = {}));
var UpdateSecurityProfileRequest;
(function (UpdateSecurityProfileRequest) {
    UpdateSecurityProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateSecurityProfileRequest = exports.UpdateSecurityProfileRequest || (exports.UpdateSecurityProfileRequest = {}));
var UpdateUserHierarchyRequest;
(function (UpdateUserHierarchyRequest) {
    UpdateUserHierarchyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateUserHierarchyRequest = exports.UpdateUserHierarchyRequest || (exports.UpdateUserHierarchyRequest = {}));
var UpdateUserHierarchyGroupNameRequest;
(function (UpdateUserHierarchyGroupNameRequest) {
    UpdateUserHierarchyGroupNameRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateUserHierarchyGroupNameRequest = exports.UpdateUserHierarchyGroupNameRequest || (exports.UpdateUserHierarchyGroupNameRequest = {}));
var HierarchyLevelUpdate;
(function (HierarchyLevelUpdate) {
    HierarchyLevelUpdate.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(HierarchyLevelUpdate = exports.HierarchyLevelUpdate || (exports.HierarchyLevelUpdate = {}));
var HierarchyStructureUpdate;
(function (HierarchyStructureUpdate) {
    HierarchyStructureUpdate.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(HierarchyStructureUpdate = exports.HierarchyStructureUpdate || (exports.HierarchyStructureUpdate = {}));
var UpdateUserHierarchyStructureRequest;
(function (UpdateUserHierarchyStructureRequest) {
    UpdateUserHierarchyStructureRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateUserHierarchyStructureRequest = exports.UpdateUserHierarchyStructureRequest || (exports.UpdateUserHierarchyStructureRequest = {}));
var UpdateUserIdentityInfoRequest;
(function (UpdateUserIdentityInfoRequest) {
    UpdateUserIdentityInfoRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateUserIdentityInfoRequest = exports.UpdateUserIdentityInfoRequest || (exports.UpdateUserIdentityInfoRequest = {}));
var UpdateUserPhoneConfigRequest;
(function (UpdateUserPhoneConfigRequest) {
    UpdateUserPhoneConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateUserPhoneConfigRequest = exports.UpdateUserPhoneConfigRequest || (exports.UpdateUserPhoneConfigRequest = {}));
var UpdateUserRoutingProfileRequest;
(function (UpdateUserRoutingProfileRequest) {
    UpdateUserRoutingProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateUserRoutingProfileRequest = exports.UpdateUserRoutingProfileRequest || (exports.UpdateUserRoutingProfileRequest = {}));
var UpdateUserSecurityProfilesRequest;
(function (UpdateUserSecurityProfilesRequest) {
    UpdateUserSecurityProfilesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateUserSecurityProfilesRequest = exports.UpdateUserSecurityProfilesRequest || (exports.UpdateUserSecurityProfilesRequest = {}));
