// @aws-sdk/client-connect/dist-es/commands/AssociateApprovedOriginCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AssociateApprovedOriginRequest } from "../models/models_0";
import { deserializeAws_restJson1AssociateApprovedOriginCommand, serializeAws_restJson1AssociateApprovedOriginCommand, } from "../protocols/Aws_restJson1";
var AssociateApprovedOriginCommand = (function (_super) {
    __extends(AssociateApprovedOriginCommand, _super);
    function AssociateApprovedOriginCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AssociateApprovedOriginCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "AssociateApprovedOriginCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AssociateApprovedOriginRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AssociateApprovedOriginCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1AssociateApprovedOriginCommand(input, context);
    };
    AssociateApprovedOriginCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1AssociateApprovedOriginCommand(output, context);
    };
    return AssociateApprovedOriginCommand;
}($Command));
export { AssociateApprovedOriginCommand };
