// @aws-sdk/client-connect/dist-es/commands/AssociateInstanceStorageConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AssociateInstanceStorageConfigRequest, AssociateInstanceStorageConfigResponse } from "../models/models_0";
import { deserializeAws_restJson1AssociateInstanceStorageConfigCommand, serializeAws_restJson1AssociateInstanceStorageConfigCommand, } from "../protocols/Aws_restJson1";
var AssociateInstanceStorageConfigCommand = (function (_super) {
    __extends(AssociateInstanceStorageConfigCommand, _super);
    function AssociateInstanceStorageConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AssociateInstanceStorageConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "AssociateInstanceStorageConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AssociateInstanceStorageConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AssociateInstanceStorageConfigResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AssociateInstanceStorageConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1AssociateInstanceStorageConfigCommand(input, context);
    };
    AssociateInstanceStorageConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1AssociateInstanceStorageConfigCommand(output, context);
    };
    return AssociateInstanceStorageConfigCommand;
}($Command));
export { AssociateInstanceStorageConfigCommand };
