// @aws-sdk/client-connect/dist-es/commands/AssociateDefaultVocabularyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AssociateDefaultVocabularyRequest, AssociateDefaultVocabularyResponse } from "../models/models_0";
import { deserializeAws_restJson1AssociateDefaultVocabularyCommand, serializeAws_restJson1AssociateDefaultVocabularyCommand, } from "../protocols/Aws_restJson1";
var AssociateDefaultVocabularyCommand = (function (_super) {
    __extends(AssociateDefaultVocabularyCommand, _super);
    function AssociateDefaultVocabularyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AssociateDefaultVocabularyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "AssociateDefaultVocabularyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AssociateDefaultVocabularyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AssociateDefaultVocabularyResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AssociateDefaultVocabularyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1AssociateDefaultVocabularyCommand(input, context);
    };
    AssociateDefaultVocabularyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1AssociateDefaultVocabularyCommand(output, context);
    };
    return AssociateDefaultVocabularyCommand;
}($Command));
export { AssociateDefaultVocabularyCommand };
