// @aws-sdk/client-connect/dist-es/commands/CreateContactFlowModuleCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateContactFlowModuleRequest, CreateContactFlowModuleResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateContactFlowModuleCommand, serializeAws_restJson1CreateContactFlowModuleCommand, } from "../protocols/Aws_restJson1";
var CreateContactFlowModuleCommand = (function (_super) {
    __extends(CreateContactFlowModuleCommand, _super);
    function CreateContactFlowModuleCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateContactFlowModuleCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "CreateContactFlowModuleCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateContactFlowModuleRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateContactFlowModuleResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateContactFlowModuleCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateContactFlowModuleCommand(input, context);
    };
    CreateContactFlowModuleCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateContactFlowModuleCommand(output, context);
    };
    return CreateContactFlowModuleCommand;
}($Command));
export { CreateContactFlowModuleCommand };
