// @aws-sdk/client-connect/dist-es/commands/CreateContactFlowCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateContactFlowRequest, CreateContactFlowResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateContactFlowCommand, serializeAws_restJson1CreateContactFlowCommand, } from "../protocols/Aws_restJson1";
var CreateContactFlowCommand = (function (_super) {
    __extends(CreateContactFlowCommand, _super);
    function CreateContactFlowCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateContactFlowCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "CreateContactFlowCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateContactFlowRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateContactFlowResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateContactFlowCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateContactFlowCommand(input, context);
    };
    CreateContactFlowCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateContactFlowCommand(output, context);
    };
    return CreateContactFlowCommand;
}($Command));
export { CreateContactFlowCommand };
