// @aws-sdk/client-connect/dist-es/commands/CreateUserHierarchyGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateUserHierarchyGroupRequest, CreateUserHierarchyGroupResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateUserHierarchyGroupCommand, serializeAws_restJson1CreateUserHierarchyGroupCommand, } from "../protocols/Aws_restJson1";
var CreateUserHierarchyGroupCommand = (function (_super) {
    __extends(CreateUserHierarchyGroupCommand, _super);
    function CreateUserHierarchyGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateUserHierarchyGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "CreateUserHierarchyGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateUserHierarchyGroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateUserHierarchyGroupResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateUserHierarchyGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateUserHierarchyGroupCommand(input, context);
    };
    CreateUserHierarchyGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateUserHierarchyGroupCommand(output, context);
    };
    return CreateUserHierarchyGroupCommand;
}($Command));
export { CreateUserHierarchyGroupCommand };
