// @aws-sdk/client-connect/dist-es/commands/CreateIntegrationAssociationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateIntegrationAssociationRequest, CreateIntegrationAssociationResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateIntegrationAssociationCommand, serializeAws_restJson1CreateIntegrationAssociationCommand, } from "../protocols/Aws_restJson1";
var CreateIntegrationAssociationCommand = (function (_super) {
    __extends(CreateIntegrationAssociationCommand, _super);
    function CreateIntegrationAssociationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateIntegrationAssociationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "CreateIntegrationAssociationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateIntegrationAssociationRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateIntegrationAssociationResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateIntegrationAssociationCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateIntegrationAssociationCommand(input, context);
    };
    CreateIntegrationAssociationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateIntegrationAssociationCommand(output, context);
    };
    return CreateIntegrationAssociationCommand;
}($Command));
export { CreateIntegrationAssociationCommand };
