// @aws-sdk/client-connect/dist-es/commands/CreateAgentStatusCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateAgentStatusRequest, CreateAgentStatusResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateAgentStatusCommand, serializeAws_restJson1CreateAgentStatusCommand, } from "../protocols/Aws_restJson1";
var CreateAgentStatusCommand = (function (_super) {
    __extends(CreateAgentStatusCommand, _super);
    function CreateAgentStatusCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateAgentStatusCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "CreateAgentStatusCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateAgentStatusRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateAgentStatusResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateAgentStatusCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateAgentStatusCommand(input, context);
    };
    CreateAgentStatusCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateAgentStatusCommand(output, context);
    };
    return CreateAgentStatusCommand;
}($Command));
export { CreateAgentStatusCommand };
