// @aws-sdk/client-connect/dist-es/commands/CreateQuickConnectCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateQuickConnectRequest, CreateQuickConnectResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateQuickConnectCommand, serializeAws_restJson1CreateQuickConnectCommand, } from "../protocols/Aws_restJson1";
var CreateQuickConnectCommand = (function (_super) {
    __extends(CreateQuickConnectCommand, _super);
    function CreateQuickConnectCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateQuickConnectCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "CreateQuickConnectCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateQuickConnectRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateQuickConnectResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateQuickConnectCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateQuickConnectCommand(input, context);
    };
    CreateQuickConnectCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateQuickConnectCommand(output, context);
    };
    return CreateQuickConnectCommand;
}($Command));
export { CreateQuickConnectCommand };
