// @aws-sdk/client-connect/dist-es/commands/DeleteContactFlowCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteContactFlowRequest } from "../models/models_0";
import { deserializeAws_restJson1DeleteContactFlowCommand, serializeAws_restJson1DeleteContactFlowCommand, } from "../protocols/Aws_restJson1";
var DeleteContactFlowCommand = (function (_super) {
    __extends(DeleteContactFlowCommand, _super);
    function DeleteContactFlowCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteContactFlowCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "DeleteContactFlowCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteContactFlowRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteContactFlowCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteContactFlowCommand(input, context);
    };
    DeleteContactFlowCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteContactFlowCommand(output, context);
    };
    return DeleteContactFlowCommand;
}($Command));
export { DeleteContactFlowCommand };
