// @aws-sdk/client-connect/dist-es/commands/DeleteIntegrationAssociationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteIntegrationAssociationRequest } from "../models/models_0";
import { deserializeAws_restJson1DeleteIntegrationAssociationCommand, serializeAws_restJson1DeleteIntegrationAssociationCommand, } from "../protocols/Aws_restJson1";
var DeleteIntegrationAssociationCommand = (function (_super) {
    __extends(DeleteIntegrationAssociationCommand, _super);
    function DeleteIntegrationAssociationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteIntegrationAssociationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "DeleteIntegrationAssociationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteIntegrationAssociationRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteIntegrationAssociationCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteIntegrationAssociationCommand(input, context);
    };
    DeleteIntegrationAssociationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteIntegrationAssociationCommand(output, context);
    };
    return DeleteIntegrationAssociationCommand;
}($Command));
export { DeleteIntegrationAssociationCommand };
