// @aws-sdk/client-connect/dist-es/commands/DeleteVocabularyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteVocabularyRequest, DeleteVocabularyResponse } from "../models/models_0";
import { deserializeAws_restJson1DeleteVocabularyCommand, serializeAws_restJson1DeleteVocabularyCommand, } from "../protocols/Aws_restJson1";
var DeleteVocabularyCommand = (function (_super) {
    __extends(DeleteVocabularyCommand, _super);
    function DeleteVocabularyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteVocabularyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "DeleteVocabularyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteVocabularyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteVocabularyResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteVocabularyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteVocabularyCommand(input, context);
    };
    DeleteVocabularyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteVocabularyCommand(output, context);
    };
    return DeleteVocabularyCommand;
}($Command));
export { DeleteVocabularyCommand };
