// @aws-sdk/client-connect/dist-es/commands/DescribeInstanceStorageConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeInstanceStorageConfigRequest, DescribeInstanceStorageConfigResponse } from "../models/models_0";
import { deserializeAws_restJson1DescribeInstanceStorageConfigCommand, serializeAws_restJson1DescribeInstanceStorageConfigCommand, } from "../protocols/Aws_restJson1";
var DescribeInstanceStorageConfigCommand = (function (_super) {
    __extends(DescribeInstanceStorageConfigCommand, _super);
    function DescribeInstanceStorageConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeInstanceStorageConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "DescribeInstanceStorageConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeInstanceStorageConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeInstanceStorageConfigResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeInstanceStorageConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeInstanceStorageConfigCommand(input, context);
    };
    DescribeInstanceStorageConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeInstanceStorageConfigCommand(output, context);
    };
    return DescribeInstanceStorageConfigCommand;
}($Command));
export { DescribeInstanceStorageConfigCommand };
