// @aws-sdk/client-connect/dist-es/commands/DescribeUserHierarchyStructureCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeUserHierarchyStructureRequest, DescribeUserHierarchyStructureResponse } from "../models/models_0";
import { deserializeAws_restJson1DescribeUserHierarchyStructureCommand, serializeAws_restJson1DescribeUserHierarchyStructureCommand, } from "../protocols/Aws_restJson1";
var DescribeUserHierarchyStructureCommand = (function (_super) {
    __extends(DescribeUserHierarchyStructureCommand, _super);
    function DescribeUserHierarchyStructureCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeUserHierarchyStructureCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "DescribeUserHierarchyStructureCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeUserHierarchyStructureRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeUserHierarchyStructureResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeUserHierarchyStructureCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeUserHierarchyStructureCommand(input, context);
    };
    DescribeUserHierarchyStructureCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeUserHierarchyStructureCommand(output, context);
    };
    return DescribeUserHierarchyStructureCommand;
}($Command));
export { DescribeUserHierarchyStructureCommand };
