// @aws-sdk/client-connect/dist-es/commands/DescribeUserHierarchyGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeUserHierarchyGroupRequest, DescribeUserHierarchyGroupResponse } from "../models/models_0";
import { deserializeAws_restJson1DescribeUserHierarchyGroupCommand, serializeAws_restJson1DescribeUserHierarchyGroupCommand, } from "../protocols/Aws_restJson1";
var DescribeUserHierarchyGroupCommand = (function (_super) {
    __extends(DescribeUserHierarchyGroupCommand, _super);
    function DescribeUserHierarchyGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeUserHierarchyGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "DescribeUserHierarchyGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeUserHierarchyGroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeUserHierarchyGroupResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeUserHierarchyGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeUserHierarchyGroupCommand(input, context);
    };
    DescribeUserHierarchyGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeUserHierarchyGroupCommand(output, context);
    };
    return DescribeUserHierarchyGroupCommand;
}($Command));
export { DescribeUserHierarchyGroupCommand };
