// @aws-sdk/client-connect/dist-es/commands/DescribeInstanceAttributeCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeInstanceAttributeRequest, DescribeInstanceAttributeResponse } from "../models/models_0";
import { deserializeAws_restJson1DescribeInstanceAttributeCommand, serializeAws_restJson1DescribeInstanceAttributeCommand, } from "../protocols/Aws_restJson1";
var DescribeInstanceAttributeCommand = (function (_super) {
    __extends(DescribeInstanceAttributeCommand, _super);
    function DescribeInstanceAttributeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeInstanceAttributeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "DescribeInstanceAttributeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeInstanceAttributeRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeInstanceAttributeResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeInstanceAttributeCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeInstanceAttributeCommand(input, context);
    };
    DescribeInstanceAttributeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeInstanceAttributeCommand(output, context);
    };
    return DescribeInstanceAttributeCommand;
}($Command));
export { DescribeInstanceAttributeCommand };
