// @aws-sdk/client-connect/dist-es/commands/DescribeUserCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeUserRequest, DescribeUserResponse } from "../models/models_0";
import { deserializeAws_restJson1DescribeUserCommand, serializeAws_restJson1DescribeUserCommand, } from "../protocols/Aws_restJson1";
var DescribeUserCommand = (function (_super) {
    __extends(DescribeUserCommand, _super);
    function DescribeUserCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeUserCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "DescribeUserCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeUserRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeUserResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeUserCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeUserCommand(input, context);
    };
    DescribeUserCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeUserCommand(output, context);
    };
    return DescribeUserCommand;
}($Command));
export { DescribeUserCommand };
