// @aws-sdk/client-connect/dist-es/commands/DescribeContactFlowModuleCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeContactFlowModuleRequest, DescribeContactFlowModuleResponse } from "../models/models_0";
import { deserializeAws_restJson1DescribeContactFlowModuleCommand, serializeAws_restJson1DescribeContactFlowModuleCommand, } from "../protocols/Aws_restJson1";
var DescribeContactFlowModuleCommand = (function (_super) {
    __extends(DescribeContactFlowModuleCommand, _super);
    function DescribeContactFlowModuleCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeContactFlowModuleCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "DescribeContactFlowModuleCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeContactFlowModuleRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeContactFlowModuleResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeContactFlowModuleCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeContactFlowModuleCommand(input, context);
    };
    DescribeContactFlowModuleCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeContactFlowModuleCommand(output, context);
    };
    return DescribeContactFlowModuleCommand;
}($Command));
export { DescribeContactFlowModuleCommand };
