// @aws-sdk/client-connect/dist-es/commands/DescribeQuickConnectCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeQuickConnectRequest, DescribeQuickConnectResponse } from "../models/models_0";
import { deserializeAws_restJson1DescribeQuickConnectCommand, serializeAws_restJson1DescribeQuickConnectCommand, } from "../protocols/Aws_restJson1";
var DescribeQuickConnectCommand = (function (_super) {
    __extends(DescribeQuickConnectCommand, _super);
    function DescribeQuickConnectCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeQuickConnectCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "DescribeQuickConnectCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeQuickConnectRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeQuickConnectResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeQuickConnectCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeQuickConnectCommand(input, context);
    };
    DescribeQuickConnectCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeQuickConnectCommand(output, context);
    };
    return DescribeQuickConnectCommand;
}($Command));
export { DescribeQuickConnectCommand };
