// @aws-sdk/client-connect/dist-es/commands/DescribeHoursOfOperationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeHoursOfOperationRequest, DescribeHoursOfOperationResponse } from "../models/models_0";
import { deserializeAws_restJson1DescribeHoursOfOperationCommand, serializeAws_restJson1DescribeHoursOfOperationCommand, } from "../protocols/Aws_restJson1";
var DescribeHoursOfOperationCommand = (function (_super) {
    __extends(DescribeHoursOfOperationCommand, _super);
    function DescribeHoursOfOperationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeHoursOfOperationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "DescribeHoursOfOperationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeHoursOfOperationRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeHoursOfOperationResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeHoursOfOperationCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeHoursOfOperationCommand(input, context);
    };
    DescribeHoursOfOperationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeHoursOfOperationCommand(output, context);
    };
    return DescribeHoursOfOperationCommand;
}($Command));
export { DescribeHoursOfOperationCommand };
