// @aws-sdk/client-connect/dist-es/commands/DisassociateQueueQuickConnectsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DisassociateQueueQuickConnectsRequest } from "../models/models_0";
import { deserializeAws_restJson1DisassociateQueueQuickConnectsCommand, serializeAws_restJson1DisassociateQueueQuickConnectsCommand, } from "../protocols/Aws_restJson1";
var DisassociateQueueQuickConnectsCommand = (function (_super) {
    __extends(DisassociateQueueQuickConnectsCommand, _super);
    function DisassociateQueueQuickConnectsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DisassociateQueueQuickConnectsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "DisassociateQueueQuickConnectsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DisassociateQueueQuickConnectsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DisassociateQueueQuickConnectsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DisassociateQueueQuickConnectsCommand(input, context);
    };
    DisassociateQueueQuickConnectsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DisassociateQueueQuickConnectsCommand(output, context);
    };
    return DisassociateQueueQuickConnectsCommand;
}($Command));
export { DisassociateQueueQuickConnectsCommand };
