// @aws-sdk/client-connect/dist-es/commands/DisassociateApprovedOriginCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DisassociateApprovedOriginRequest } from "../models/models_0";
import { deserializeAws_restJson1DisassociateApprovedOriginCommand, serializeAws_restJson1DisassociateApprovedOriginCommand, } from "../protocols/Aws_restJson1";
var DisassociateApprovedOriginCommand = (function (_super) {
    __extends(DisassociateApprovedOriginCommand, _super);
    function DisassociateApprovedOriginCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DisassociateApprovedOriginCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "DisassociateApprovedOriginCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DisassociateApprovedOriginRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DisassociateApprovedOriginCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DisassociateApprovedOriginCommand(input, context);
    };
    DisassociateApprovedOriginCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DisassociateApprovedOriginCommand(output, context);
    };
    return DisassociateApprovedOriginCommand;
}($Command));
export { DisassociateApprovedOriginCommand };
