// @aws-sdk/client-connect/dist-es/commands/DisassociateInstanceStorageConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DisassociateInstanceStorageConfigRequest } from "../models/models_0";
import { deserializeAws_restJson1DisassociateInstanceStorageConfigCommand, serializeAws_restJson1DisassociateInstanceStorageConfigCommand, } from "../protocols/Aws_restJson1";
var DisassociateInstanceStorageConfigCommand = (function (_super) {
    __extends(DisassociateInstanceStorageConfigCommand, _super);
    function DisassociateInstanceStorageConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DisassociateInstanceStorageConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "DisassociateInstanceStorageConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DisassociateInstanceStorageConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DisassociateInstanceStorageConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DisassociateInstanceStorageConfigCommand(input, context);
    };
    DisassociateInstanceStorageConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DisassociateInstanceStorageConfigCommand(output, context);
    };
    return DisassociateInstanceStorageConfigCommand;
}($Command));
export { DisassociateInstanceStorageConfigCommand };
