// @aws-sdk/client-connect/dist-es/commands/GetMetricDataCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetMetricDataRequest, GetMetricDataResponse } from "../models/models_0";
import { deserializeAws_restJson1GetMetricDataCommand, serializeAws_restJson1GetMetricDataCommand, } from "../protocols/Aws_restJson1";
var GetMetricDataCommand = (function (_super) {
    __extends(GetMetricDataCommand, _super);
    function GetMetricDataCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetMetricDataCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "GetMetricDataCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetMetricDataRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetMetricDataResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetMetricDataCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetMetricDataCommand(input, context);
    };
    GetMetricDataCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetMetricDataCommand(output, context);
    };
    return GetMetricDataCommand;
}($Command));
export { GetMetricDataCommand };
