// @aws-sdk/client-connect/dist-es/commands/ListQueueQuickConnectsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListQueueQuickConnectsRequest, ListQueueQuickConnectsResponse } from "../models/models_0";
import { deserializeAws_restJson1ListQueueQuickConnectsCommand, serializeAws_restJson1ListQueueQuickConnectsCommand, } from "../protocols/Aws_restJson1";
var ListQueueQuickConnectsCommand = (function (_super) {
    __extends(ListQueueQuickConnectsCommand, _super);
    function ListQueueQuickConnectsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListQueueQuickConnectsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "ListQueueQuickConnectsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListQueueQuickConnectsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListQueueQuickConnectsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListQueueQuickConnectsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListQueueQuickConnectsCommand(input, context);
    };
    ListQueueQuickConnectsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListQueueQuickConnectsCommand(output, context);
    };
    return ListQueueQuickConnectsCommand;
}($Command));
export { ListQueueQuickConnectsCommand };
