// @aws-sdk/client-connect/dist-es/commands/ListLexBotsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListLexBotsRequest, ListLexBotsResponse } from "../models/models_0";
import { deserializeAws_restJson1ListLexBotsCommand, serializeAws_restJson1ListLexBotsCommand, } from "../protocols/Aws_restJson1";
var ListLexBotsCommand = (function (_super) {
    __extends(ListLexBotsCommand, _super);
    function ListLexBotsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListLexBotsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "ListLexBotsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListLexBotsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListLexBotsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListLexBotsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListLexBotsCommand(input, context);
    };
    ListLexBotsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListLexBotsCommand(output, context);
    };
    return ListLexBotsCommand;
}($Command));
export { ListLexBotsCommand };
