// @aws-sdk/client-connect/dist-es/commands/ListContactReferencesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListContactReferencesRequest, ListContactReferencesResponse } from "../models/models_0";
import { deserializeAws_restJson1ListContactReferencesCommand, serializeAws_restJson1ListContactReferencesCommand, } from "../protocols/Aws_restJson1";
var ListContactReferencesCommand = (function (_super) {
    __extends(ListContactReferencesCommand, _super);
    function ListContactReferencesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListContactReferencesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "ListContactReferencesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListContactReferencesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListContactReferencesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListContactReferencesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListContactReferencesCommand(input, context);
    };
    ListContactReferencesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListContactReferencesCommand(output, context);
    };
    return ListContactReferencesCommand;
}($Command));
export { ListContactReferencesCommand };
