// @aws-sdk/client-connect/dist-es/commands/ListBotsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListBotsRequest, ListBotsResponse } from "../models/models_0";
import { deserializeAws_restJson1ListBotsCommand, serializeAws_restJson1ListBotsCommand, } from "../protocols/Aws_restJson1";
var ListBotsCommand = (function (_super) {
    __extends(ListBotsCommand, _super);
    function ListBotsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListBotsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "ListBotsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListBotsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListBotsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListBotsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListBotsCommand(input, context);
    };
    ListBotsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListBotsCommand(output, context);
    };
    return ListBotsCommand;
}($Command));
export { ListBotsCommand };
