// @aws-sdk/client-connect/dist-es/commands/ListQueuesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListQueuesRequest, ListQueuesResponse } from "../models/models_0";
import { deserializeAws_restJson1ListQueuesCommand, serializeAws_restJson1ListQueuesCommand, } from "../protocols/Aws_restJson1";
var ListQueuesCommand = (function (_super) {
    __extends(ListQueuesCommand, _super);
    function ListQueuesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListQueuesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "ListQueuesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListQueuesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListQueuesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListQueuesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListQueuesCommand(input, context);
    };
    ListQueuesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListQueuesCommand(output, context);
    };
    return ListQueuesCommand;
}($Command));
export { ListQueuesCommand };
