// @aws-sdk/client-connect/dist-es/commands/ListSecurityKeysCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListSecurityKeysRequest, ListSecurityKeysResponse } from "../models/models_0";
import { deserializeAws_restJson1ListSecurityKeysCommand, serializeAws_restJson1ListSecurityKeysCommand, } from "../protocols/Aws_restJson1";
var ListSecurityKeysCommand = (function (_super) {
    __extends(ListSecurityKeysCommand, _super);
    function ListSecurityKeysCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListSecurityKeysCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "ListSecurityKeysCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListSecurityKeysRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListSecurityKeysResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListSecurityKeysCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListSecurityKeysCommand(input, context);
    };
    ListSecurityKeysCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListSecurityKeysCommand(output, context);
    };
    return ListSecurityKeysCommand;
}($Command));
export { ListSecurityKeysCommand };
