// @aws-sdk/client-connect/dist-es/commands/ListPhoneNumbersCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListPhoneNumbersRequest, ListPhoneNumbersResponse } from "../models/models_0";
import { deserializeAws_restJson1ListPhoneNumbersCommand, serializeAws_restJson1ListPhoneNumbersCommand, } from "../protocols/Aws_restJson1";
var ListPhoneNumbersCommand = (function (_super) {
    __extends(ListPhoneNumbersCommand, _super);
    function ListPhoneNumbersCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListPhoneNumbersCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "ListPhoneNumbersCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListPhoneNumbersRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListPhoneNumbersResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListPhoneNumbersCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListPhoneNumbersCommand(input, context);
    };
    ListPhoneNumbersCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListPhoneNumbersCommand(output, context);
    };
    return ListPhoneNumbersCommand;
}($Command));
export { ListPhoneNumbersCommand };
