// @aws-sdk/client-connect/dist-es/commands/ListDefaultVocabulariesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListDefaultVocabulariesRequest, ListDefaultVocabulariesResponse } from "../models/models_0";
import { deserializeAws_restJson1ListDefaultVocabulariesCommand, serializeAws_restJson1ListDefaultVocabulariesCommand, } from "../protocols/Aws_restJson1";
var ListDefaultVocabulariesCommand = (function (_super) {
    __extends(ListDefaultVocabulariesCommand, _super);
    function ListDefaultVocabulariesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListDefaultVocabulariesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "ListDefaultVocabulariesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListDefaultVocabulariesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListDefaultVocabulariesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListDefaultVocabulariesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListDefaultVocabulariesCommand(input, context);
    };
    ListDefaultVocabulariesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListDefaultVocabulariesCommand(output, context);
    };
    return ListDefaultVocabulariesCommand;
}($Command));
export { ListDefaultVocabulariesCommand };
