// @aws-sdk/client-connect/dist-es/commands/SearchVocabulariesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SearchVocabulariesRequest, SearchVocabulariesResponse } from "../models/models_0";
import { deserializeAws_restJson1SearchVocabulariesCommand, serializeAws_restJson1SearchVocabulariesCommand, } from "../protocols/Aws_restJson1";
var SearchVocabulariesCommand = (function (_super) {
    __extends(SearchVocabulariesCommand, _super);
    function SearchVocabulariesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    SearchVocabulariesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "SearchVocabulariesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: SearchVocabulariesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: SearchVocabulariesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    SearchVocabulariesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1SearchVocabulariesCommand(input, context);
    };
    SearchVocabulariesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1SearchVocabulariesCommand(output, context);
    };
    return SearchVocabulariesCommand;
}($Command));
export { SearchVocabulariesCommand };
