// @aws-sdk/client-connect/dist-es/commands/StartOutboundVoiceContactCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartOutboundVoiceContactRequest, StartOutboundVoiceContactResponse } from "../models/models_1";
import { deserializeAws_restJson1StartOutboundVoiceContactCommand, serializeAws_restJson1StartOutboundVoiceContactCommand, } from "../protocols/Aws_restJson1";
var StartOutboundVoiceContactCommand = (function (_super) {
    __extends(StartOutboundVoiceContactCommand, _super);
    function StartOutboundVoiceContactCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StartOutboundVoiceContactCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "StartOutboundVoiceContactCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StartOutboundVoiceContactRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StartOutboundVoiceContactResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StartOutboundVoiceContactCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1StartOutboundVoiceContactCommand(input, context);
    };
    StartOutboundVoiceContactCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1StartOutboundVoiceContactCommand(output, context);
    };
    return StartOutboundVoiceContactCommand;
}($Command));
export { StartOutboundVoiceContactCommand };
