// @aws-sdk/client-connect/dist-es/commands/StartChatContactCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartChatContactRequest, StartChatContactResponse } from "../models/models_0";
import { deserializeAws_restJson1StartChatContactCommand, serializeAws_restJson1StartChatContactCommand, } from "../protocols/Aws_restJson1";
var StartChatContactCommand = (function (_super) {
    __extends(StartChatContactCommand, _super);
    function StartChatContactCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StartChatContactCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "StartChatContactCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StartChatContactRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StartChatContactResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StartChatContactCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1StartChatContactCommand(input, context);
    };
    StartChatContactCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1StartChatContactCommand(output, context);
    };
    return StartChatContactCommand;
}($Command));
export { StartChatContactCommand };
