// @aws-sdk/client-connect/dist-es/commands/StartTaskContactCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartTaskContactRequest, StartTaskContactResponse } from "../models/models_1";
import { deserializeAws_restJson1StartTaskContactCommand, serializeAws_restJson1StartTaskContactCommand, } from "../protocols/Aws_restJson1";
var StartTaskContactCommand = (function (_super) {
    __extends(StartTaskContactCommand, _super);
    function StartTaskContactCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StartTaskContactCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "StartTaskContactCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StartTaskContactRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StartTaskContactResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StartTaskContactCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1StartTaskContactCommand(input, context);
    };
    StartTaskContactCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1StartTaskContactCommand(output, context);
    };
    return StartTaskContactCommand;
}($Command));
export { StartTaskContactCommand };
