// @aws-sdk/client-connect/dist-es/commands/StartContactRecordingCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartContactRecordingRequest, StartContactRecordingResponse } from "../models/models_0";
import { deserializeAws_restJson1StartContactRecordingCommand, serializeAws_restJson1StartContactRecordingCommand, } from "../protocols/Aws_restJson1";
var StartContactRecordingCommand = (function (_super) {
    __extends(StartContactRecordingCommand, _super);
    function StartContactRecordingCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StartContactRecordingCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "StartContactRecordingCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StartContactRecordingRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StartContactRecordingResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StartContactRecordingCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1StartContactRecordingCommand(input, context);
    };
    StartContactRecordingCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1StartContactRecordingCommand(output, context);
    };
    return StartContactRecordingCommand;
}($Command));
export { StartContactRecordingCommand };
