// @aws-sdk/client-connect/dist-es/commands/StartContactStreamingCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartContactStreamingRequest, StartContactStreamingResponse } from "../models/models_0";
import { deserializeAws_restJson1StartContactStreamingCommand, serializeAws_restJson1StartContactStreamingCommand, } from "../protocols/Aws_restJson1";
var StartContactStreamingCommand = (function (_super) {
    __extends(StartContactStreamingCommand, _super);
    function StartContactStreamingCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StartContactStreamingCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "StartContactStreamingCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StartContactStreamingRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StartContactStreamingResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StartContactStreamingCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1StartContactStreamingCommand(input, context);
    };
    StartContactStreamingCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1StartContactStreamingCommand(output, context);
    };
    return StartContactStreamingCommand;
}($Command));
export { StartContactStreamingCommand };
