// @aws-sdk/client-connect/dist-es/commands/StopContactStreamingCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StopContactStreamingRequest, StopContactStreamingResponse } from "../models/models_1";
import { deserializeAws_restJson1StopContactStreamingCommand, serializeAws_restJson1StopContactStreamingCommand, } from "../protocols/Aws_restJson1";
var StopContactStreamingCommand = (function (_super) {
    __extends(StopContactStreamingCommand, _super);
    function StopContactStreamingCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StopContactStreamingCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "StopContactStreamingCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StopContactStreamingRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StopContactStreamingResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StopContactStreamingCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1StopContactStreamingCommand(input, context);
    };
    StopContactStreamingCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1StopContactStreamingCommand(output, context);
    };
    return StopContactStreamingCommand;
}($Command));
export { StopContactStreamingCommand };
