// @aws-sdk/client-connect/dist-es/commands/StopContactCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StopContactRequest, StopContactResponse } from "../models/models_1";
import { deserializeAws_restJson1StopContactCommand, serializeAws_restJson1StopContactCommand, } from "../protocols/Aws_restJson1";
var StopContactCommand = (function (_super) {
    __extends(StopContactCommand, _super);
    function StopContactCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StopContactCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "StopContactCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StopContactRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StopContactResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StopContactCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1StopContactCommand(input, context);
    };
    StopContactCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1StopContactCommand(output, context);
    };
    return StopContactCommand;
}($Command));
export { StopContactCommand };
