// @aws-sdk/client-connect/dist-es/commands/UpdateContactFlowModuleMetadataCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateContactFlowModuleMetadataRequest, UpdateContactFlowModuleMetadataResponse } from "../models/models_1";
import { deserializeAws_restJson1UpdateContactFlowModuleMetadataCommand, serializeAws_restJson1UpdateContactFlowModuleMetadataCommand, } from "../protocols/Aws_restJson1";
var UpdateContactFlowModuleMetadataCommand = (function (_super) {
    __extends(UpdateContactFlowModuleMetadataCommand, _super);
    function UpdateContactFlowModuleMetadataCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateContactFlowModuleMetadataCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "UpdateContactFlowModuleMetadataCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateContactFlowModuleMetadataRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateContactFlowModuleMetadataResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateContactFlowModuleMetadataCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateContactFlowModuleMetadataCommand(input, context);
    };
    UpdateContactFlowModuleMetadataCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateContactFlowModuleMetadataCommand(output, context);
    };
    return UpdateContactFlowModuleMetadataCommand;
}($Command));
export { UpdateContactFlowModuleMetadataCommand };
