// @aws-sdk/client-connect/dist-es/commands/UpdateContactAttributesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateContactAttributesRequest, UpdateContactAttributesResponse } from "../models/models_1";
import { deserializeAws_restJson1UpdateContactAttributesCommand, serializeAws_restJson1UpdateContactAttributesCommand, } from "../protocols/Aws_restJson1";
var UpdateContactAttributesCommand = (function (_super) {
    __extends(UpdateContactAttributesCommand, _super);
    function UpdateContactAttributesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateContactAttributesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "UpdateContactAttributesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateContactAttributesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateContactAttributesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateContactAttributesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateContactAttributesCommand(input, context);
    };
    UpdateContactAttributesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateContactAttributesCommand(output, context);
    };
    return UpdateContactAttributesCommand;
}($Command));
export { UpdateContactAttributesCommand };
