// @aws-sdk/client-connect/dist-es/commands/UpdateContactFlowNameCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateContactFlowNameRequest } from "../models/models_1";
import { deserializeAws_restJson1UpdateContactFlowNameCommand, serializeAws_restJson1UpdateContactFlowNameCommand, } from "../protocols/Aws_restJson1";
var UpdateContactFlowNameCommand = (function (_super) {
    __extends(UpdateContactFlowNameCommand, _super);
    function UpdateContactFlowNameCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateContactFlowNameCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "UpdateContactFlowNameCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateContactFlowNameRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateContactFlowNameCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateContactFlowNameCommand(input, context);
    };
    UpdateContactFlowNameCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateContactFlowNameCommand(output, context);
    };
    return UpdateContactFlowNameCommand;
}($Command));
export { UpdateContactFlowNameCommand };
