// @aws-sdk/client-connect/dist-es/commands/UpdateContactScheduleCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateContactScheduleRequest, UpdateContactScheduleResponse } from "../models/models_1";
import { deserializeAws_restJson1UpdateContactScheduleCommand, serializeAws_restJson1UpdateContactScheduleCommand, } from "../protocols/Aws_restJson1";
var UpdateContactScheduleCommand = (function (_super) {
    __extends(UpdateContactScheduleCommand, _super);
    function UpdateContactScheduleCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateContactScheduleCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "UpdateContactScheduleCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateContactScheduleRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateContactScheduleResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateContactScheduleCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateContactScheduleCommand(input, context);
    };
    UpdateContactScheduleCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateContactScheduleCommand(output, context);
    };
    return UpdateContactScheduleCommand;
}($Command));
export { UpdateContactScheduleCommand };
