// @aws-sdk/client-connect/dist-es/commands/UpdateQueueStatusCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateQueueStatusRequest } from "../models/models_1";
import { deserializeAws_restJson1UpdateQueueStatusCommand, serializeAws_restJson1UpdateQueueStatusCommand, } from "../protocols/Aws_restJson1";
var UpdateQueueStatusCommand = (function (_super) {
    __extends(UpdateQueueStatusCommand, _super);
    function UpdateQueueStatusCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateQueueStatusCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "UpdateQueueStatusCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateQueueStatusRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateQueueStatusCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateQueueStatusCommand(input, context);
    };
    UpdateQueueStatusCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateQueueStatusCommand(output, context);
    };
    return UpdateQueueStatusCommand;
}($Command));
export { UpdateQueueStatusCommand };
