// @aws-sdk/client-connect/dist-es/commands/UpdateUserPhoneConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateUserPhoneConfigRequest } from "../models/models_1";
import { deserializeAws_restJson1UpdateUserPhoneConfigCommand, serializeAws_restJson1UpdateUserPhoneConfigCommand, } from "../protocols/Aws_restJson1";
var UpdateUserPhoneConfigCommand = (function (_super) {
    __extends(UpdateUserPhoneConfigCommand, _super);
    function UpdateUserPhoneConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateUserPhoneConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "UpdateUserPhoneConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateUserPhoneConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateUserPhoneConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateUserPhoneConfigCommand(input, context);
    };
    UpdateUserPhoneConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateUserPhoneConfigCommand(output, context);
    };
    return UpdateUserPhoneConfigCommand;
}($Command));
export { UpdateUserPhoneConfigCommand };
