// @aws-sdk/client-connect/dist-es/commands/UpdateContactFlowContentCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateContactFlowContentRequest } from "../models/models_1";
import { deserializeAws_restJson1UpdateContactFlowContentCommand, serializeAws_restJson1UpdateContactFlowContentCommand, } from "../protocols/Aws_restJson1";
var UpdateContactFlowContentCommand = (function (_super) {
    __extends(UpdateContactFlowContentCommand, _super);
    function UpdateContactFlowContentCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateContactFlowContentCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "UpdateContactFlowContentCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateContactFlowContentRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateContactFlowContentCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateContactFlowContentCommand(input, context);
    };
    UpdateContactFlowContentCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateContactFlowContentCommand(output, context);
    };
    return UpdateContactFlowContentCommand;
}($Command));
export { UpdateContactFlowContentCommand };
