// @aws-sdk/client-connect/dist-es/commands/UpdateHoursOfOperationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateHoursOfOperationRequest } from "../models/models_1";
import { deserializeAws_restJson1UpdateHoursOfOperationCommand, serializeAws_restJson1UpdateHoursOfOperationCommand, } from "../protocols/Aws_restJson1";
var UpdateHoursOfOperationCommand = (function (_super) {
    __extends(UpdateHoursOfOperationCommand, _super);
    function UpdateHoursOfOperationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateHoursOfOperationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "UpdateHoursOfOperationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateHoursOfOperationRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateHoursOfOperationCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateHoursOfOperationCommand(input, context);
    };
    UpdateHoursOfOperationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateHoursOfOperationCommand(output, context);
    };
    return UpdateHoursOfOperationCommand;
}($Command));
export { UpdateHoursOfOperationCommand };
