// @aws-sdk/client-connect/dist-es/commands/UpdateQuickConnectConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateQuickConnectConfigRequest } from "../models/models_1";
import { deserializeAws_restJson1UpdateQuickConnectConfigCommand, serializeAws_restJson1UpdateQuickConnectConfigCommand, } from "../protocols/Aws_restJson1";
var UpdateQuickConnectConfigCommand = (function (_super) {
    __extends(UpdateQuickConnectConfigCommand, _super);
    function UpdateQuickConnectConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateQuickConnectConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "UpdateQuickConnectConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateQuickConnectConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateQuickConnectConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateQuickConnectConfigCommand(input, context);
    };
    UpdateQuickConnectConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateQuickConnectConfigCommand(output, context);
    };
    return UpdateQuickConnectConfigCommand;
}($Command));
export { UpdateQuickConnectConfigCommand };
