// @aws-sdk/client-connect/dist-es/commands/UpdateRoutingProfileNameCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateRoutingProfileNameRequest } from "../models/models_1";
import { deserializeAws_restJson1UpdateRoutingProfileNameCommand, serializeAws_restJson1UpdateRoutingProfileNameCommand, } from "../protocols/Aws_restJson1";
var UpdateRoutingProfileNameCommand = (function (_super) {
    __extends(UpdateRoutingProfileNameCommand, _super);
    function UpdateRoutingProfileNameCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateRoutingProfileNameCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ConnectClient";
        var commandName = "UpdateRoutingProfileNameCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateRoutingProfileNameRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateRoutingProfileNameCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateRoutingProfileNameCommand(input, context);
    };
    UpdateRoutingProfileNameCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateRoutingProfileNameCommand(output, context);
    };
    return UpdateRoutingProfileNameCommand;
}($Command));
export { UpdateRoutingProfileNameCommand };
