// @aws-sdk/client-connect/dist-es/models/models_0.js
import { __assign } from "tslib";
import { SENSITIVE_STRING } from "@aws-sdk/smithy-client";
export var AgentInfo;
(function (AgentInfo) {
    AgentInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AgentInfo || (AgentInfo = {}));
export var AgentStatusState;
(function (AgentStatusState) {
    AgentStatusState["DISABLED"] = "DISABLED";
    AgentStatusState["ENABLED"] = "ENABLED";
})(AgentStatusState || (AgentStatusState = {}));
export var AgentStatusType;
(function (AgentStatusType) {
    AgentStatusType["CUSTOM"] = "CUSTOM";
    AgentStatusType["OFFLINE"] = "OFFLINE";
    AgentStatusType["ROUTABLE"] = "ROUTABLE";
})(AgentStatusType || (AgentStatusType = {}));
export var AgentStatus;
(function (AgentStatus) {
    AgentStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AgentStatus || (AgentStatus = {}));
export var AgentStatusSummary;
(function (AgentStatusSummary) {
    AgentStatusSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AgentStatusSummary || (AgentStatusSummary = {}));
export var AssociateApprovedOriginRequest;
(function (AssociateApprovedOriginRequest) {
    AssociateApprovedOriginRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociateApprovedOriginRequest || (AssociateApprovedOriginRequest = {}));
export var LexBot;
(function (LexBot) {
    LexBot.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LexBot || (LexBot = {}));
export var LexV2Bot;
(function (LexV2Bot) {
    LexV2Bot.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LexV2Bot || (LexV2Bot = {}));
export var AssociateBotRequest;
(function (AssociateBotRequest) {
    AssociateBotRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociateBotRequest || (AssociateBotRequest = {}));
export var VocabularyLanguageCode;
(function (VocabularyLanguageCode) {
    VocabularyLanguageCode["AR_AE"] = "ar-AE";
    VocabularyLanguageCode["DE_CH"] = "de-CH";
    VocabularyLanguageCode["DE_DE"] = "de-DE";
    VocabularyLanguageCode["EN_AB"] = "en-AB";
    VocabularyLanguageCode["EN_AU"] = "en-AU";
    VocabularyLanguageCode["EN_GB"] = "en-GB";
    VocabularyLanguageCode["EN_IE"] = "en-IE";
    VocabularyLanguageCode["EN_IN"] = "en-IN";
    VocabularyLanguageCode["EN_US"] = "en-US";
    VocabularyLanguageCode["EN_WL"] = "en-WL";
    VocabularyLanguageCode["ES_ES"] = "es-ES";
    VocabularyLanguageCode["ES_US"] = "es-US";
    VocabularyLanguageCode["FR_CA"] = "fr-CA";
    VocabularyLanguageCode["FR_FR"] = "fr-FR";
    VocabularyLanguageCode["HI_IN"] = "hi-IN";
    VocabularyLanguageCode["IT_IT"] = "it-IT";
    VocabularyLanguageCode["JA_JP"] = "ja-JP";
    VocabularyLanguageCode["KO_KR"] = "ko-KR";
    VocabularyLanguageCode["PT_BR"] = "pt-BR";
    VocabularyLanguageCode["PT_PT"] = "pt-PT";
    VocabularyLanguageCode["ZH_CN"] = "zh-CN";
})(VocabularyLanguageCode || (VocabularyLanguageCode = {}));
export var AssociateDefaultVocabularyRequest;
(function (AssociateDefaultVocabularyRequest) {
    AssociateDefaultVocabularyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociateDefaultVocabularyRequest || (AssociateDefaultVocabularyRequest = {}));
export var AssociateDefaultVocabularyResponse;
(function (AssociateDefaultVocabularyResponse) {
    AssociateDefaultVocabularyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociateDefaultVocabularyResponse || (AssociateDefaultVocabularyResponse = {}));
export var InstanceStorageResourceType;
(function (InstanceStorageResourceType) {
    InstanceStorageResourceType["AGENT_EVENTS"] = "AGENT_EVENTS";
    InstanceStorageResourceType["CALL_RECORDINGS"] = "CALL_RECORDINGS";
    InstanceStorageResourceType["CHAT_TRANSCRIPTS"] = "CHAT_TRANSCRIPTS";
    InstanceStorageResourceType["CONTACT_TRACE_RECORDS"] = "CONTACT_TRACE_RECORDS";
    InstanceStorageResourceType["MEDIA_STREAMS"] = "MEDIA_STREAMS";
    InstanceStorageResourceType["SCHEDULED_REPORTS"] = "SCHEDULED_REPORTS";
})(InstanceStorageResourceType || (InstanceStorageResourceType = {}));
export var KinesisFirehoseConfig;
(function (KinesisFirehoseConfig) {
    KinesisFirehoseConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(KinesisFirehoseConfig || (KinesisFirehoseConfig = {}));
export var KinesisStreamConfig;
(function (KinesisStreamConfig) {
    KinesisStreamConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(KinesisStreamConfig || (KinesisStreamConfig = {}));
export var EncryptionType;
(function (EncryptionType) {
    EncryptionType["KMS"] = "KMS";
})(EncryptionType || (EncryptionType = {}));
export var EncryptionConfig;
(function (EncryptionConfig) {
    EncryptionConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EncryptionConfig || (EncryptionConfig = {}));
export var KinesisVideoStreamConfig;
(function (KinesisVideoStreamConfig) {
    KinesisVideoStreamConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(KinesisVideoStreamConfig || (KinesisVideoStreamConfig = {}));
export var S3Config;
(function (S3Config) {
    S3Config.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(S3Config || (S3Config = {}));
export var StorageType;
(function (StorageType) {
    StorageType["KINESIS_FIREHOSE"] = "KINESIS_FIREHOSE";
    StorageType["KINESIS_STREAM"] = "KINESIS_STREAM";
    StorageType["KINESIS_VIDEO_STREAM"] = "KINESIS_VIDEO_STREAM";
    StorageType["S3"] = "S3";
})(StorageType || (StorageType = {}));
export var InstanceStorageConfig;
(function (InstanceStorageConfig) {
    InstanceStorageConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InstanceStorageConfig || (InstanceStorageConfig = {}));
export var AssociateInstanceStorageConfigRequest;
(function (AssociateInstanceStorageConfigRequest) {
    AssociateInstanceStorageConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociateInstanceStorageConfigRequest || (AssociateInstanceStorageConfigRequest = {}));
export var AssociateInstanceStorageConfigResponse;
(function (AssociateInstanceStorageConfigResponse) {
    AssociateInstanceStorageConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociateInstanceStorageConfigResponse || (AssociateInstanceStorageConfigResponse = {}));
export var AssociateLambdaFunctionRequest;
(function (AssociateLambdaFunctionRequest) {
    AssociateLambdaFunctionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociateLambdaFunctionRequest || (AssociateLambdaFunctionRequest = {}));
export var AssociateLexBotRequest;
(function (AssociateLexBotRequest) {
    AssociateLexBotRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociateLexBotRequest || (AssociateLexBotRequest = {}));
export var AssociateQueueQuickConnectsRequest;
(function (AssociateQueueQuickConnectsRequest) {
    AssociateQueueQuickConnectsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociateQueueQuickConnectsRequest || (AssociateQueueQuickConnectsRequest = {}));
export var Channel;
(function (Channel) {
    Channel["CHAT"] = "CHAT";
    Channel["TASK"] = "TASK";
    Channel["VOICE"] = "VOICE";
})(Channel || (Channel = {}));
export var RoutingProfileQueueReference;
(function (RoutingProfileQueueReference) {
    RoutingProfileQueueReference.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RoutingProfileQueueReference || (RoutingProfileQueueReference = {}));
export var RoutingProfileQueueConfig;
(function (RoutingProfileQueueConfig) {
    RoutingProfileQueueConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RoutingProfileQueueConfig || (RoutingProfileQueueConfig = {}));
export var AssociateRoutingProfileQueuesRequest;
(function (AssociateRoutingProfileQueuesRequest) {
    AssociateRoutingProfileQueuesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociateRoutingProfileQueuesRequest || (AssociateRoutingProfileQueuesRequest = {}));
export var AssociateSecurityKeyRequest;
(function (AssociateSecurityKeyRequest) {
    AssociateSecurityKeyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociateSecurityKeyRequest || (AssociateSecurityKeyRequest = {}));
export var AssociateSecurityKeyResponse;
(function (AssociateSecurityKeyResponse) {
    AssociateSecurityKeyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociateSecurityKeyResponse || (AssociateSecurityKeyResponse = {}));
export var CreateAgentStatusRequest;
(function (CreateAgentStatusRequest) {
    CreateAgentStatusRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateAgentStatusRequest || (CreateAgentStatusRequest = {}));
export var CreateAgentStatusResponse;
(function (CreateAgentStatusResponse) {
    CreateAgentStatusResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateAgentStatusResponse || (CreateAgentStatusResponse = {}));
export var ContactFlowType;
(function (ContactFlowType) {
    ContactFlowType["AGENT_HOLD"] = "AGENT_HOLD";
    ContactFlowType["AGENT_TRANSFER"] = "AGENT_TRANSFER";
    ContactFlowType["AGENT_WHISPER"] = "AGENT_WHISPER";
    ContactFlowType["CONTACT_FLOW"] = "CONTACT_FLOW";
    ContactFlowType["CUSTOMER_HOLD"] = "CUSTOMER_HOLD";
    ContactFlowType["CUSTOMER_QUEUE"] = "CUSTOMER_QUEUE";
    ContactFlowType["CUSTOMER_WHISPER"] = "CUSTOMER_WHISPER";
    ContactFlowType["OUTBOUND_WHISPER"] = "OUTBOUND_WHISPER";
    ContactFlowType["QUEUE_TRANSFER"] = "QUEUE_TRANSFER";
})(ContactFlowType || (ContactFlowType = {}));
export var CreateContactFlowRequest;
(function (CreateContactFlowRequest) {
    CreateContactFlowRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateContactFlowRequest || (CreateContactFlowRequest = {}));
export var CreateContactFlowResponse;
(function (CreateContactFlowResponse) {
    CreateContactFlowResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateContactFlowResponse || (CreateContactFlowResponse = {}));
export var ProblemDetail;
(function (ProblemDetail) {
    ProblemDetail.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ProblemDetail || (ProblemDetail = {}));
export var CreateContactFlowModuleRequest;
(function (CreateContactFlowModuleRequest) {
    CreateContactFlowModuleRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateContactFlowModuleRequest || (CreateContactFlowModuleRequest = {}));
export var CreateContactFlowModuleResponse;
(function (CreateContactFlowModuleResponse) {
    CreateContactFlowModuleResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateContactFlowModuleResponse || (CreateContactFlowModuleResponse = {}));
export var HoursOfOperationDays;
(function (HoursOfOperationDays) {
    HoursOfOperationDays["FRIDAY"] = "FRIDAY";
    HoursOfOperationDays["MONDAY"] = "MONDAY";
    HoursOfOperationDays["SATURDAY"] = "SATURDAY";
    HoursOfOperationDays["SUNDAY"] = "SUNDAY";
    HoursOfOperationDays["THURSDAY"] = "THURSDAY";
    HoursOfOperationDays["TUESDAY"] = "TUESDAY";
    HoursOfOperationDays["WEDNESDAY"] = "WEDNESDAY";
})(HoursOfOperationDays || (HoursOfOperationDays = {}));
export var HoursOfOperationTimeSlice;
(function (HoursOfOperationTimeSlice) {
    HoursOfOperationTimeSlice.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(HoursOfOperationTimeSlice || (HoursOfOperationTimeSlice = {}));
export var HoursOfOperationConfig;
(function (HoursOfOperationConfig) {
    HoursOfOperationConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(HoursOfOperationConfig || (HoursOfOperationConfig = {}));
export var CreateHoursOfOperationRequest;
(function (CreateHoursOfOperationRequest) {
    CreateHoursOfOperationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateHoursOfOperationRequest || (CreateHoursOfOperationRequest = {}));
export var CreateHoursOfOperationResponse;
(function (CreateHoursOfOperationResponse) {
    CreateHoursOfOperationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateHoursOfOperationResponse || (CreateHoursOfOperationResponse = {}));
export var DirectoryType;
(function (DirectoryType) {
    DirectoryType["CONNECT_MANAGED"] = "CONNECT_MANAGED";
    DirectoryType["EXISTING_DIRECTORY"] = "EXISTING_DIRECTORY";
    DirectoryType["SAML"] = "SAML";
})(DirectoryType || (DirectoryType = {}));
export var CreateInstanceRequest;
(function (CreateInstanceRequest) {
    CreateInstanceRequest.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.InstanceAlias && { InstanceAlias: SENSITIVE_STRING }))); };
})(CreateInstanceRequest || (CreateInstanceRequest = {}));
export var CreateInstanceResponse;
(function (CreateInstanceResponse) {
    CreateInstanceResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateInstanceResponse || (CreateInstanceResponse = {}));
export var IntegrationType;
(function (IntegrationType) {
    IntegrationType["EVENT"] = "EVENT";
    IntegrationType["PINPOINT_APP"] = "PINPOINT_APP";
    IntegrationType["VOICE_ID"] = "VOICE_ID";
    IntegrationType["WISDOM_ASSISTANT"] = "WISDOM_ASSISTANT";
    IntegrationType["WISDOM_KNOWLEDGE_BASE"] = "WISDOM_KNOWLEDGE_BASE";
})(IntegrationType || (IntegrationType = {}));
export var SourceType;
(function (SourceType) {
    SourceType["SALESFORCE"] = "SALESFORCE";
    SourceType["ZENDESK"] = "ZENDESK";
})(SourceType || (SourceType = {}));
export var CreateIntegrationAssociationRequest;
(function (CreateIntegrationAssociationRequest) {
    CreateIntegrationAssociationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateIntegrationAssociationRequest || (CreateIntegrationAssociationRequest = {}));
export var CreateIntegrationAssociationResponse;
(function (CreateIntegrationAssociationResponse) {
    CreateIntegrationAssociationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateIntegrationAssociationResponse || (CreateIntegrationAssociationResponse = {}));
export var OutboundCallerConfig;
(function (OutboundCallerConfig) {
    OutboundCallerConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OutboundCallerConfig || (OutboundCallerConfig = {}));
export var CreateQueueRequest;
(function (CreateQueueRequest) {
    CreateQueueRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateQueueRequest || (CreateQueueRequest = {}));
export var CreateQueueResponse;
(function (CreateQueueResponse) {
    CreateQueueResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateQueueResponse || (CreateQueueResponse = {}));
export var PhoneNumberQuickConnectConfig;
(function (PhoneNumberQuickConnectConfig) {
    PhoneNumberQuickConnectConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PhoneNumberQuickConnectConfig || (PhoneNumberQuickConnectConfig = {}));
export var QueueQuickConnectConfig;
(function (QueueQuickConnectConfig) {
    QueueQuickConnectConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(QueueQuickConnectConfig || (QueueQuickConnectConfig = {}));
export var QuickConnectType;
(function (QuickConnectType) {
    QuickConnectType["PHONE_NUMBER"] = "PHONE_NUMBER";
    QuickConnectType["QUEUE"] = "QUEUE";
    QuickConnectType["USER"] = "USER";
})(QuickConnectType || (QuickConnectType = {}));
export var UserQuickConnectConfig;
(function (UserQuickConnectConfig) {
    UserQuickConnectConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UserQuickConnectConfig || (UserQuickConnectConfig = {}));
export var QuickConnectConfig;
(function (QuickConnectConfig) {
    QuickConnectConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(QuickConnectConfig || (QuickConnectConfig = {}));
export var CreateQuickConnectRequest;
(function (CreateQuickConnectRequest) {
    CreateQuickConnectRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateQuickConnectRequest || (CreateQuickConnectRequest = {}));
export var CreateQuickConnectResponse;
(function (CreateQuickConnectResponse) {
    CreateQuickConnectResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateQuickConnectResponse || (CreateQuickConnectResponse = {}));
export var MediaConcurrency;
(function (MediaConcurrency) {
    MediaConcurrency.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MediaConcurrency || (MediaConcurrency = {}));
export var CreateRoutingProfileRequest;
(function (CreateRoutingProfileRequest) {
    CreateRoutingProfileRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateRoutingProfileRequest || (CreateRoutingProfileRequest = {}));
export var CreateRoutingProfileResponse;
(function (CreateRoutingProfileResponse) {
    CreateRoutingProfileResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateRoutingProfileResponse || (CreateRoutingProfileResponse = {}));
export var CreateSecurityProfileRequest;
(function (CreateSecurityProfileRequest) {
    CreateSecurityProfileRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateSecurityProfileRequest || (CreateSecurityProfileRequest = {}));
export var CreateSecurityProfileResponse;
(function (CreateSecurityProfileResponse) {
    CreateSecurityProfileResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateSecurityProfileResponse || (CreateSecurityProfileResponse = {}));
export var UseCaseType;
(function (UseCaseType) {
    UseCaseType["CONNECT_CAMPAIGNS"] = "CONNECT_CAMPAIGNS";
    UseCaseType["RULES_EVALUATION"] = "RULES_EVALUATION";
})(UseCaseType || (UseCaseType = {}));
export var CreateUseCaseRequest;
(function (CreateUseCaseRequest) {
    CreateUseCaseRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateUseCaseRequest || (CreateUseCaseRequest = {}));
export var CreateUseCaseResponse;
(function (CreateUseCaseResponse) {
    CreateUseCaseResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateUseCaseResponse || (CreateUseCaseResponse = {}));
export var UserIdentityInfo;
(function (UserIdentityInfo) {
    UserIdentityInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UserIdentityInfo || (UserIdentityInfo = {}));
export var PhoneType;
(function (PhoneType) {
    PhoneType["DESK_PHONE"] = "DESK_PHONE";
    PhoneType["SOFT_PHONE"] = "SOFT_PHONE";
})(PhoneType || (PhoneType = {}));
export var UserPhoneConfig;
(function (UserPhoneConfig) {
    UserPhoneConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UserPhoneConfig || (UserPhoneConfig = {}));
export var CreateUserRequest;
(function (CreateUserRequest) {
    CreateUserRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateUserRequest || (CreateUserRequest = {}));
export var CreateUserResponse;
(function (CreateUserResponse) {
    CreateUserResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateUserResponse || (CreateUserResponse = {}));
export var CreateUserHierarchyGroupRequest;
(function (CreateUserHierarchyGroupRequest) {
    CreateUserHierarchyGroupRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateUserHierarchyGroupRequest || (CreateUserHierarchyGroupRequest = {}));
export var CreateUserHierarchyGroupResponse;
(function (CreateUserHierarchyGroupResponse) {
    CreateUserHierarchyGroupResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateUserHierarchyGroupResponse || (CreateUserHierarchyGroupResponse = {}));
export var CreateVocabularyRequest;
(function (CreateVocabularyRequest) {
    CreateVocabularyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateVocabularyRequest || (CreateVocabularyRequest = {}));
export var VocabularyState;
(function (VocabularyState) {
    VocabularyState["ACTIVE"] = "ACTIVE";
    VocabularyState["CREATION_FAILED"] = "CREATION_FAILED";
    VocabularyState["CREATION_IN_PROGRESS"] = "CREATION_IN_PROGRESS";
    VocabularyState["DELETE_IN_PROGRESS"] = "DELETE_IN_PROGRESS";
})(VocabularyState || (VocabularyState = {}));
export var CreateVocabularyResponse;
(function (CreateVocabularyResponse) {
    CreateVocabularyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateVocabularyResponse || (CreateVocabularyResponse = {}));
export var DeleteContactFlowRequest;
(function (DeleteContactFlowRequest) {
    DeleteContactFlowRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteContactFlowRequest || (DeleteContactFlowRequest = {}));
export var DeleteContactFlowModuleRequest;
(function (DeleteContactFlowModuleRequest) {
    DeleteContactFlowModuleRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteContactFlowModuleRequest || (DeleteContactFlowModuleRequest = {}));
export var DeleteContactFlowModuleResponse;
(function (DeleteContactFlowModuleResponse) {
    DeleteContactFlowModuleResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteContactFlowModuleResponse || (DeleteContactFlowModuleResponse = {}));
export var DeleteHoursOfOperationRequest;
(function (DeleteHoursOfOperationRequest) {
    DeleteHoursOfOperationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteHoursOfOperationRequest || (DeleteHoursOfOperationRequest = {}));
export var DeleteInstanceRequest;
(function (DeleteInstanceRequest) {
    DeleteInstanceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteInstanceRequest || (DeleteInstanceRequest = {}));
export var DeleteIntegrationAssociationRequest;
(function (DeleteIntegrationAssociationRequest) {
    DeleteIntegrationAssociationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteIntegrationAssociationRequest || (DeleteIntegrationAssociationRequest = {}));
export var DeleteQuickConnectRequest;
(function (DeleteQuickConnectRequest) {
    DeleteQuickConnectRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteQuickConnectRequest || (DeleteQuickConnectRequest = {}));
export var DeleteSecurityProfileRequest;
(function (DeleteSecurityProfileRequest) {
    DeleteSecurityProfileRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteSecurityProfileRequest || (DeleteSecurityProfileRequest = {}));
export var ResourceType;
(function (ResourceType) {
    ResourceType["CONTACT"] = "CONTACT";
    ResourceType["CONTACT_FLOW"] = "CONTACT_FLOW";
    ResourceType["HIERARCHY_GROUP"] = "HIERARCHY_GROUP";
    ResourceType["HIERARCHY_LEVEL"] = "HIERARCHY_LEVEL";
    ResourceType["INSTANCE"] = "INSTANCE";
    ResourceType["PARTICIPANT"] = "PARTICIPANT";
    ResourceType["USER"] = "USER";
})(ResourceType || (ResourceType = {}));
export var DeleteUseCaseRequest;
(function (DeleteUseCaseRequest) {
    DeleteUseCaseRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteUseCaseRequest || (DeleteUseCaseRequest = {}));
export var DeleteUserRequest;
(function (DeleteUserRequest) {
    DeleteUserRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteUserRequest || (DeleteUserRequest = {}));
export var DeleteUserHierarchyGroupRequest;
(function (DeleteUserHierarchyGroupRequest) {
    DeleteUserHierarchyGroupRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteUserHierarchyGroupRequest || (DeleteUserHierarchyGroupRequest = {}));
export var DeleteVocabularyRequest;
(function (DeleteVocabularyRequest) {
    DeleteVocabularyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteVocabularyRequest || (DeleteVocabularyRequest = {}));
export var DeleteVocabularyResponse;
(function (DeleteVocabularyResponse) {
    DeleteVocabularyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteVocabularyResponse || (DeleteVocabularyResponse = {}));
export var DescribeAgentStatusRequest;
(function (DescribeAgentStatusRequest) {
    DescribeAgentStatusRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAgentStatusRequest || (DescribeAgentStatusRequest = {}));
export var DescribeAgentStatusResponse;
(function (DescribeAgentStatusResponse) {
    DescribeAgentStatusResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAgentStatusResponse || (DescribeAgentStatusResponse = {}));
export var DescribeContactRequest;
(function (DescribeContactRequest) {
    DescribeContactRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeContactRequest || (DescribeContactRequest = {}));
export var ContactInitiationMethod;
(function (ContactInitiationMethod) {
    ContactInitiationMethod["API"] = "API";
    ContactInitiationMethod["CALLBACK"] = "CALLBACK";
    ContactInitiationMethod["INBOUND"] = "INBOUND";
    ContactInitiationMethod["OUTBOUND"] = "OUTBOUND";
    ContactInitiationMethod["QUEUE_TRANSFER"] = "QUEUE_TRANSFER";
    ContactInitiationMethod["TRANSFER"] = "TRANSFER";
})(ContactInitiationMethod || (ContactInitiationMethod = {}));
export var QueueInfo;
(function (QueueInfo) {
    QueueInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(QueueInfo || (QueueInfo = {}));
export var Contact;
(function (Contact) {
    Contact.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Contact || (Contact = {}));
export var DescribeContactResponse;
(function (DescribeContactResponse) {
    DescribeContactResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeContactResponse || (DescribeContactResponse = {}));
export var DescribeContactFlowRequest;
(function (DescribeContactFlowRequest) {
    DescribeContactFlowRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeContactFlowRequest || (DescribeContactFlowRequest = {}));
export var ContactFlowState;
(function (ContactFlowState) {
    ContactFlowState["ACTIVE"] = "ACTIVE";
    ContactFlowState["ARCHIVED"] = "ARCHIVED";
})(ContactFlowState || (ContactFlowState = {}));
export var ContactFlow;
(function (ContactFlow) {
    ContactFlow.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ContactFlow || (ContactFlow = {}));
export var DescribeContactFlowResponse;
(function (DescribeContactFlowResponse) {
    DescribeContactFlowResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeContactFlowResponse || (DescribeContactFlowResponse = {}));
export var DescribeContactFlowModuleRequest;
(function (DescribeContactFlowModuleRequest) {
    DescribeContactFlowModuleRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeContactFlowModuleRequest || (DescribeContactFlowModuleRequest = {}));
export var ContactFlowModuleState;
(function (ContactFlowModuleState) {
    ContactFlowModuleState["ACTIVE"] = "ACTIVE";
    ContactFlowModuleState["ARCHIVED"] = "ARCHIVED";
})(ContactFlowModuleState || (ContactFlowModuleState = {}));
export var ContactFlowModuleStatus;
(function (ContactFlowModuleStatus) {
    ContactFlowModuleStatus["PUBLISHED"] = "PUBLISHED";
    ContactFlowModuleStatus["SAVED"] = "SAVED";
})(ContactFlowModuleStatus || (ContactFlowModuleStatus = {}));
export var ContactFlowModule;
(function (ContactFlowModule) {
    ContactFlowModule.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ContactFlowModule || (ContactFlowModule = {}));
export var DescribeContactFlowModuleResponse;
(function (DescribeContactFlowModuleResponse) {
    DescribeContactFlowModuleResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeContactFlowModuleResponse || (DescribeContactFlowModuleResponse = {}));
export var DescribeHoursOfOperationRequest;
(function (DescribeHoursOfOperationRequest) {
    DescribeHoursOfOperationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeHoursOfOperationRequest || (DescribeHoursOfOperationRequest = {}));
export var HoursOfOperation;
(function (HoursOfOperation) {
    HoursOfOperation.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(HoursOfOperation || (HoursOfOperation = {}));
export var DescribeHoursOfOperationResponse;
(function (DescribeHoursOfOperationResponse) {
    DescribeHoursOfOperationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeHoursOfOperationResponse || (DescribeHoursOfOperationResponse = {}));
export var DescribeInstanceRequest;
(function (DescribeInstanceRequest) {
    DescribeInstanceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeInstanceRequest || (DescribeInstanceRequest = {}));
export var InstanceStatus;
(function (InstanceStatus) {
    InstanceStatus["ACTIVE"] = "ACTIVE";
    InstanceStatus["CREATION_FAILED"] = "CREATION_FAILED";
    InstanceStatus["CREATION_IN_PROGRESS"] = "CREATION_IN_PROGRESS";
})(InstanceStatus || (InstanceStatus = {}));
export var InstanceStatusReason;
(function (InstanceStatusReason) {
    InstanceStatusReason.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InstanceStatusReason || (InstanceStatusReason = {}));
export var Instance;
(function (Instance) {
    Instance.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.InstanceAlias && { InstanceAlias: SENSITIVE_STRING }))); };
})(Instance || (Instance = {}));
export var DescribeInstanceResponse;
(function (DescribeInstanceResponse) {
    DescribeInstanceResponse.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Instance && { Instance: Instance.filterSensitiveLog(obj.Instance) }))); };
})(DescribeInstanceResponse || (DescribeInstanceResponse = {}));
export var InstanceAttributeType;
(function (InstanceAttributeType) {
    InstanceAttributeType["AUTO_RESOLVE_BEST_VOICES"] = "AUTO_RESOLVE_BEST_VOICES";
    InstanceAttributeType["CONTACTFLOW_LOGS"] = "CONTACTFLOW_LOGS";
    InstanceAttributeType["CONTACT_LENS"] = "CONTACT_LENS";
    InstanceAttributeType["EARLY_MEDIA"] = "EARLY_MEDIA";
    InstanceAttributeType["INBOUND_CALLS"] = "INBOUND_CALLS";
    InstanceAttributeType["OUTBOUND_CALLS"] = "OUTBOUND_CALLS";
    InstanceAttributeType["USE_CUSTOM_TTS_VOICES"] = "USE_CUSTOM_TTS_VOICES";
})(InstanceAttributeType || (InstanceAttributeType = {}));
export var DescribeInstanceAttributeRequest;
(function (DescribeInstanceAttributeRequest) {
    DescribeInstanceAttributeRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeInstanceAttributeRequest || (DescribeInstanceAttributeRequest = {}));
export var Attribute;
(function (Attribute) {
    Attribute.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Attribute || (Attribute = {}));
export var DescribeInstanceAttributeResponse;
(function (DescribeInstanceAttributeResponse) {
    DescribeInstanceAttributeResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeInstanceAttributeResponse || (DescribeInstanceAttributeResponse = {}));
export var DescribeInstanceStorageConfigRequest;
(function (DescribeInstanceStorageConfigRequest) {
    DescribeInstanceStorageConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeInstanceStorageConfigRequest || (DescribeInstanceStorageConfigRequest = {}));
export var DescribeInstanceStorageConfigResponse;
(function (DescribeInstanceStorageConfigResponse) {
    DescribeInstanceStorageConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeInstanceStorageConfigResponse || (DescribeInstanceStorageConfigResponse = {}));
export var DescribeQueueRequest;
(function (DescribeQueueRequest) {
    DescribeQueueRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeQueueRequest || (DescribeQueueRequest = {}));
export var QueueStatus;
(function (QueueStatus) {
    QueueStatus["DISABLED"] = "DISABLED";
    QueueStatus["ENABLED"] = "ENABLED";
})(QueueStatus || (QueueStatus = {}));
export var Queue;
(function (Queue) {
    Queue.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Queue || (Queue = {}));
export var DescribeQueueResponse;
(function (DescribeQueueResponse) {
    DescribeQueueResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeQueueResponse || (DescribeQueueResponse = {}));
export var DescribeQuickConnectRequest;
(function (DescribeQuickConnectRequest) {
    DescribeQuickConnectRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeQuickConnectRequest || (DescribeQuickConnectRequest = {}));
export var QuickConnect;
(function (QuickConnect) {
    QuickConnect.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(QuickConnect || (QuickConnect = {}));
export var DescribeQuickConnectResponse;
(function (DescribeQuickConnectResponse) {
    DescribeQuickConnectResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeQuickConnectResponse || (DescribeQuickConnectResponse = {}));
export var DescribeRoutingProfileRequest;
(function (DescribeRoutingProfileRequest) {
    DescribeRoutingProfileRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeRoutingProfileRequest || (DescribeRoutingProfileRequest = {}));
export var RoutingProfile;
(function (RoutingProfile) {
    RoutingProfile.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RoutingProfile || (RoutingProfile = {}));
export var DescribeRoutingProfileResponse;
(function (DescribeRoutingProfileResponse) {
    DescribeRoutingProfileResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeRoutingProfileResponse || (DescribeRoutingProfileResponse = {}));
export var DescribeSecurityProfileRequest;
(function (DescribeSecurityProfileRequest) {
    DescribeSecurityProfileRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeSecurityProfileRequest || (DescribeSecurityProfileRequest = {}));
export var SecurityProfile;
(function (SecurityProfile) {
    SecurityProfile.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SecurityProfile || (SecurityProfile = {}));
export var DescribeSecurityProfileResponse;
(function (DescribeSecurityProfileResponse) {
    DescribeSecurityProfileResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeSecurityProfileResponse || (DescribeSecurityProfileResponse = {}));
export var DescribeUserRequest;
(function (DescribeUserRequest) {
    DescribeUserRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeUserRequest || (DescribeUserRequest = {}));
export var User;
(function (User) {
    User.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(User || (User = {}));
export var DescribeUserResponse;
(function (DescribeUserResponse) {
    DescribeUserResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeUserResponse || (DescribeUserResponse = {}));
export var DescribeUserHierarchyGroupRequest;
(function (DescribeUserHierarchyGroupRequest) {
    DescribeUserHierarchyGroupRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeUserHierarchyGroupRequest || (DescribeUserHierarchyGroupRequest = {}));
export var HierarchyGroupSummary;
(function (HierarchyGroupSummary) {
    HierarchyGroupSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(HierarchyGroupSummary || (HierarchyGroupSummary = {}));
export var HierarchyPath;
(function (HierarchyPath) {
    HierarchyPath.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(HierarchyPath || (HierarchyPath = {}));
export var HierarchyGroup;
(function (HierarchyGroup) {
    HierarchyGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(HierarchyGroup || (HierarchyGroup = {}));
export var DescribeUserHierarchyGroupResponse;
(function (DescribeUserHierarchyGroupResponse) {
    DescribeUserHierarchyGroupResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeUserHierarchyGroupResponse || (DescribeUserHierarchyGroupResponse = {}));
export var DescribeUserHierarchyStructureRequest;
(function (DescribeUserHierarchyStructureRequest) {
    DescribeUserHierarchyStructureRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeUserHierarchyStructureRequest || (DescribeUserHierarchyStructureRequest = {}));
export var HierarchyLevel;
(function (HierarchyLevel) {
    HierarchyLevel.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(HierarchyLevel || (HierarchyLevel = {}));
export var HierarchyStructure;
(function (HierarchyStructure) {
    HierarchyStructure.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(HierarchyStructure || (HierarchyStructure = {}));
export var DescribeUserHierarchyStructureResponse;
(function (DescribeUserHierarchyStructureResponse) {
    DescribeUserHierarchyStructureResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeUserHierarchyStructureResponse || (DescribeUserHierarchyStructureResponse = {}));
export var DescribeVocabularyRequest;
(function (DescribeVocabularyRequest) {
    DescribeVocabularyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeVocabularyRequest || (DescribeVocabularyRequest = {}));
export var Vocabulary;
(function (Vocabulary) {
    Vocabulary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Vocabulary || (Vocabulary = {}));
export var DescribeVocabularyResponse;
(function (DescribeVocabularyResponse) {
    DescribeVocabularyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeVocabularyResponse || (DescribeVocabularyResponse = {}));
export var DisassociateApprovedOriginRequest;
(function (DisassociateApprovedOriginRequest) {
    DisassociateApprovedOriginRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisassociateApprovedOriginRequest || (DisassociateApprovedOriginRequest = {}));
export var DisassociateBotRequest;
(function (DisassociateBotRequest) {
    DisassociateBotRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisassociateBotRequest || (DisassociateBotRequest = {}));
export var DisassociateInstanceStorageConfigRequest;
(function (DisassociateInstanceStorageConfigRequest) {
    DisassociateInstanceStorageConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisassociateInstanceStorageConfigRequest || (DisassociateInstanceStorageConfigRequest = {}));
export var DisassociateLambdaFunctionRequest;
(function (DisassociateLambdaFunctionRequest) {
    DisassociateLambdaFunctionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisassociateLambdaFunctionRequest || (DisassociateLambdaFunctionRequest = {}));
export var DisassociateLexBotRequest;
(function (DisassociateLexBotRequest) {
    DisassociateLexBotRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisassociateLexBotRequest || (DisassociateLexBotRequest = {}));
export var DisassociateQueueQuickConnectsRequest;
(function (DisassociateQueueQuickConnectsRequest) {
    DisassociateQueueQuickConnectsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisassociateQueueQuickConnectsRequest || (DisassociateQueueQuickConnectsRequest = {}));
export var DisassociateRoutingProfileQueuesRequest;
(function (DisassociateRoutingProfileQueuesRequest) {
    DisassociateRoutingProfileQueuesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisassociateRoutingProfileQueuesRequest || (DisassociateRoutingProfileQueuesRequest = {}));
export var DisassociateSecurityKeyRequest;
(function (DisassociateSecurityKeyRequest) {
    DisassociateSecurityKeyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisassociateSecurityKeyRequest || (DisassociateSecurityKeyRequest = {}));
export var GetContactAttributesRequest;
(function (GetContactAttributesRequest) {
    GetContactAttributesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetContactAttributesRequest || (GetContactAttributesRequest = {}));
export var GetContactAttributesResponse;
(function (GetContactAttributesResponse) {
    GetContactAttributesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetContactAttributesResponse || (GetContactAttributesResponse = {}));
export var CurrentMetricName;
(function (CurrentMetricName) {
    CurrentMetricName["AGENTS_AFTER_CONTACT_WORK"] = "AGENTS_AFTER_CONTACT_WORK";
    CurrentMetricName["AGENTS_AVAILABLE"] = "AGENTS_AVAILABLE";
    CurrentMetricName["AGENTS_ERROR"] = "AGENTS_ERROR";
    CurrentMetricName["AGENTS_NON_PRODUCTIVE"] = "AGENTS_NON_PRODUCTIVE";
    CurrentMetricName["AGENTS_ONLINE"] = "AGENTS_ONLINE";
    CurrentMetricName["AGENTS_ON_CALL"] = "AGENTS_ON_CALL";
    CurrentMetricName["AGENTS_ON_CONTACT"] = "AGENTS_ON_CONTACT";
    CurrentMetricName["AGENTS_STAFFED"] = "AGENTS_STAFFED";
    CurrentMetricName["CONTACTS_IN_QUEUE"] = "CONTACTS_IN_QUEUE";
    CurrentMetricName["CONTACTS_SCHEDULED"] = "CONTACTS_SCHEDULED";
    CurrentMetricName["OLDEST_CONTACT_AGE"] = "OLDEST_CONTACT_AGE";
    CurrentMetricName["SLOTS_ACTIVE"] = "SLOTS_ACTIVE";
    CurrentMetricName["SLOTS_AVAILABLE"] = "SLOTS_AVAILABLE";
})(CurrentMetricName || (CurrentMetricName = {}));
export var Unit;
(function (Unit) {
    Unit["COUNT"] = "COUNT";
    Unit["PERCENT"] = "PERCENT";
    Unit["SECONDS"] = "SECONDS";
})(Unit || (Unit = {}));
export var CurrentMetric;
(function (CurrentMetric) {
    CurrentMetric.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CurrentMetric || (CurrentMetric = {}));
export var Filters;
(function (Filters) {
    Filters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Filters || (Filters = {}));
export var Grouping;
(function (Grouping) {
    Grouping["CHANNEL"] = "CHANNEL";
    Grouping["QUEUE"] = "QUEUE";
})(Grouping || (Grouping = {}));
export var GetCurrentMetricDataRequest;
(function (GetCurrentMetricDataRequest) {
    GetCurrentMetricDataRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetCurrentMetricDataRequest || (GetCurrentMetricDataRequest = {}));
export var CurrentMetricData;
(function (CurrentMetricData) {
    CurrentMetricData.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CurrentMetricData || (CurrentMetricData = {}));
export var QueueReference;
(function (QueueReference) {
    QueueReference.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(QueueReference || (QueueReference = {}));
export var Dimensions;
(function (Dimensions) {
    Dimensions.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Dimensions || (Dimensions = {}));
export var CurrentMetricResult;
(function (CurrentMetricResult) {
    CurrentMetricResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CurrentMetricResult || (CurrentMetricResult = {}));
export var GetCurrentMetricDataResponse;
(function (GetCurrentMetricDataResponse) {
    GetCurrentMetricDataResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetCurrentMetricDataResponse || (GetCurrentMetricDataResponse = {}));
export var GetFederationTokenRequest;
(function (GetFederationTokenRequest) {
    GetFederationTokenRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetFederationTokenRequest || (GetFederationTokenRequest = {}));
export var Credentials;
(function (Credentials) {
    Credentials.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.AccessToken && { AccessToken: SENSITIVE_STRING })), (obj.RefreshToken && { RefreshToken: SENSITIVE_STRING }))); };
})(Credentials || (Credentials = {}));
export var GetFederationTokenResponse;
(function (GetFederationTokenResponse) {
    GetFederationTokenResponse.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Credentials && { Credentials: Credentials.filterSensitiveLog(obj.Credentials) }))); };
})(GetFederationTokenResponse || (GetFederationTokenResponse = {}));
export var HistoricalMetricName;
(function (HistoricalMetricName) {
    HistoricalMetricName["ABANDON_TIME"] = "ABANDON_TIME";
    HistoricalMetricName["AFTER_CONTACT_WORK_TIME"] = "AFTER_CONTACT_WORK_TIME";
    HistoricalMetricName["API_CONTACTS_HANDLED"] = "API_CONTACTS_HANDLED";
    HistoricalMetricName["CALLBACK_CONTACTS_HANDLED"] = "CALLBACK_CONTACTS_HANDLED";
    HistoricalMetricName["CONTACTS_ABANDONED"] = "CONTACTS_ABANDONED";
    HistoricalMetricName["CONTACTS_AGENT_HUNG_UP_FIRST"] = "CONTACTS_AGENT_HUNG_UP_FIRST";
    HistoricalMetricName["CONTACTS_CONSULTED"] = "CONTACTS_CONSULTED";
    HistoricalMetricName["CONTACTS_HANDLED"] = "CONTACTS_HANDLED";
    HistoricalMetricName["CONTACTS_HANDLED_INCOMING"] = "CONTACTS_HANDLED_INCOMING";
    HistoricalMetricName["CONTACTS_HANDLED_OUTBOUND"] = "CONTACTS_HANDLED_OUTBOUND";
    HistoricalMetricName["CONTACTS_HOLD_ABANDONS"] = "CONTACTS_HOLD_ABANDONS";
    HistoricalMetricName["CONTACTS_MISSED"] = "CONTACTS_MISSED";
    HistoricalMetricName["CONTACTS_QUEUED"] = "CONTACTS_QUEUED";
    HistoricalMetricName["CONTACTS_TRANSFERRED_IN"] = "CONTACTS_TRANSFERRED_IN";
    HistoricalMetricName["CONTACTS_TRANSFERRED_IN_FROM_QUEUE"] = "CONTACTS_TRANSFERRED_IN_FROM_QUEUE";
    HistoricalMetricName["CONTACTS_TRANSFERRED_OUT"] = "CONTACTS_TRANSFERRED_OUT";
    HistoricalMetricName["CONTACTS_TRANSFERRED_OUT_FROM_QUEUE"] = "CONTACTS_TRANSFERRED_OUT_FROM_QUEUE";
    HistoricalMetricName["HANDLE_TIME"] = "HANDLE_TIME";
    HistoricalMetricName["HOLD_TIME"] = "HOLD_TIME";
    HistoricalMetricName["INTERACTION_AND_HOLD_TIME"] = "INTERACTION_AND_HOLD_TIME";
    HistoricalMetricName["INTERACTION_TIME"] = "INTERACTION_TIME";
    HistoricalMetricName["OCCUPANCY"] = "OCCUPANCY";
    HistoricalMetricName["QUEUED_TIME"] = "QUEUED_TIME";
    HistoricalMetricName["QUEUE_ANSWER_TIME"] = "QUEUE_ANSWER_TIME";
    HistoricalMetricName["SERVICE_LEVEL"] = "SERVICE_LEVEL";
})(HistoricalMetricName || (HistoricalMetricName = {}));
export var Statistic;
(function (Statistic) {
    Statistic["AVG"] = "AVG";
    Statistic["MAX"] = "MAX";
    Statistic["SUM"] = "SUM";
})(Statistic || (Statistic = {}));
export var Comparison;
(function (Comparison) {
    Comparison["LT"] = "LT";
})(Comparison || (Comparison = {}));
export var Threshold;
(function (Threshold) {
    Threshold.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Threshold || (Threshold = {}));
export var HistoricalMetric;
(function (HistoricalMetric) {
    HistoricalMetric.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(HistoricalMetric || (HistoricalMetric = {}));
export var GetMetricDataRequest;
(function (GetMetricDataRequest) {
    GetMetricDataRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetMetricDataRequest || (GetMetricDataRequest = {}));
export var HistoricalMetricData;
(function (HistoricalMetricData) {
    HistoricalMetricData.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(HistoricalMetricData || (HistoricalMetricData = {}));
export var HistoricalMetricResult;
(function (HistoricalMetricResult) {
    HistoricalMetricResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(HistoricalMetricResult || (HistoricalMetricResult = {}));
export var GetMetricDataResponse;
(function (GetMetricDataResponse) {
    GetMetricDataResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetMetricDataResponse || (GetMetricDataResponse = {}));
export var ListAgentStatusRequest;
(function (ListAgentStatusRequest) {
    ListAgentStatusRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAgentStatusRequest || (ListAgentStatusRequest = {}));
export var ListAgentStatusResponse;
(function (ListAgentStatusResponse) {
    ListAgentStatusResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAgentStatusResponse || (ListAgentStatusResponse = {}));
export var ListApprovedOriginsRequest;
(function (ListApprovedOriginsRequest) {
    ListApprovedOriginsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListApprovedOriginsRequest || (ListApprovedOriginsRequest = {}));
export var ListApprovedOriginsResponse;
(function (ListApprovedOriginsResponse) {
    ListApprovedOriginsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListApprovedOriginsResponse || (ListApprovedOriginsResponse = {}));
export var LexVersion;
(function (LexVersion) {
    LexVersion["V1"] = "V1";
    LexVersion["V2"] = "V2";
})(LexVersion || (LexVersion = {}));
export var ListBotsRequest;
(function (ListBotsRequest) {
    ListBotsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListBotsRequest || (ListBotsRequest = {}));
export var LexBotConfig;
(function (LexBotConfig) {
    LexBotConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LexBotConfig || (LexBotConfig = {}));
export var ListBotsResponse;
(function (ListBotsResponse) {
    ListBotsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListBotsResponse || (ListBotsResponse = {}));
export var ListContactFlowModulesRequest;
(function (ListContactFlowModulesRequest) {
    ListContactFlowModulesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListContactFlowModulesRequest || (ListContactFlowModulesRequest = {}));
export var ContactFlowModuleSummary;
(function (ContactFlowModuleSummary) {
    ContactFlowModuleSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ContactFlowModuleSummary || (ContactFlowModuleSummary = {}));
export var ListContactFlowModulesResponse;
(function (ListContactFlowModulesResponse) {
    ListContactFlowModulesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListContactFlowModulesResponse || (ListContactFlowModulesResponse = {}));
export var ListContactFlowsRequest;
(function (ListContactFlowsRequest) {
    ListContactFlowsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListContactFlowsRequest || (ListContactFlowsRequest = {}));
export var ContactFlowSummary;
(function (ContactFlowSummary) {
    ContactFlowSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ContactFlowSummary || (ContactFlowSummary = {}));
export var ListContactFlowsResponse;
(function (ListContactFlowsResponse) {
    ListContactFlowsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListContactFlowsResponse || (ListContactFlowsResponse = {}));
export var ReferenceType;
(function (ReferenceType) {
    ReferenceType["ATTACHMENT"] = "ATTACHMENT";
    ReferenceType["URL"] = "URL";
})(ReferenceType || (ReferenceType = {}));
export var ListContactReferencesRequest;
(function (ListContactReferencesRequest) {
    ListContactReferencesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListContactReferencesRequest || (ListContactReferencesRequest = {}));
export var ReferenceStatus;
(function (ReferenceStatus) {
    ReferenceStatus["APPROVED"] = "APPROVED";
    ReferenceStatus["REJECTED"] = "REJECTED";
})(ReferenceStatus || (ReferenceStatus = {}));
export var AttachmentReference;
(function (AttachmentReference) {
    AttachmentReference.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AttachmentReference || (AttachmentReference = {}));
export var UrlReference;
(function (UrlReference) {
    UrlReference.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UrlReference || (UrlReference = {}));
export var ReferenceSummary;
(function (ReferenceSummary) {
    ReferenceSummary.visit = function (value, visitor) {
        if (value.Url !== undefined)
            return visitor.Url(value.Url);
        if (value.Attachment !== undefined)
            return visitor.Attachment(value.Attachment);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
    ReferenceSummary.filterSensitiveLog = function (obj) {
        var _a;
        if (obj.Url !== undefined)
            return { Url: UrlReference.filterSensitiveLog(obj.Url) };
        if (obj.Attachment !== undefined)
            return { Attachment: AttachmentReference.filterSensitiveLog(obj.Attachment) };
        if (obj.$unknown !== undefined)
            return _a = {}, _a[obj.$unknown[0]] = "UNKNOWN", _a;
    };
})(ReferenceSummary || (ReferenceSummary = {}));
export var ListContactReferencesResponse;
(function (ListContactReferencesResponse) {
    ListContactReferencesResponse.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.ReferenceSummaryList && {
        ReferenceSummaryList: obj.ReferenceSummaryList.map(function (item) { return ReferenceSummary.filterSensitiveLog(item); }),
    }))); };
})(ListContactReferencesResponse || (ListContactReferencesResponse = {}));
export var ListDefaultVocabulariesRequest;
(function (ListDefaultVocabulariesRequest) {
    ListDefaultVocabulariesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDefaultVocabulariesRequest || (ListDefaultVocabulariesRequest = {}));
export var DefaultVocabulary;
(function (DefaultVocabulary) {
    DefaultVocabulary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DefaultVocabulary || (DefaultVocabulary = {}));
export var ListDefaultVocabulariesResponse;
(function (ListDefaultVocabulariesResponse) {
    ListDefaultVocabulariesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDefaultVocabulariesResponse || (ListDefaultVocabulariesResponse = {}));
export var ListHoursOfOperationsRequest;
(function (ListHoursOfOperationsRequest) {
    ListHoursOfOperationsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListHoursOfOperationsRequest || (ListHoursOfOperationsRequest = {}));
export var HoursOfOperationSummary;
(function (HoursOfOperationSummary) {
    HoursOfOperationSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(HoursOfOperationSummary || (HoursOfOperationSummary = {}));
export var ListHoursOfOperationsResponse;
(function (ListHoursOfOperationsResponse) {
    ListHoursOfOperationsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListHoursOfOperationsResponse || (ListHoursOfOperationsResponse = {}));
export var ListInstanceAttributesRequest;
(function (ListInstanceAttributesRequest) {
    ListInstanceAttributesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListInstanceAttributesRequest || (ListInstanceAttributesRequest = {}));
export var ListInstanceAttributesResponse;
(function (ListInstanceAttributesResponse) {
    ListInstanceAttributesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListInstanceAttributesResponse || (ListInstanceAttributesResponse = {}));
export var ListInstancesRequest;
(function (ListInstancesRequest) {
    ListInstancesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListInstancesRequest || (ListInstancesRequest = {}));
export var InstanceSummary;
(function (InstanceSummary) {
    InstanceSummary.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.InstanceAlias && { InstanceAlias: SENSITIVE_STRING }))); };
})(InstanceSummary || (InstanceSummary = {}));
export var ListInstancesResponse;
(function (ListInstancesResponse) {
    ListInstancesResponse.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.InstanceSummaryList && {
        InstanceSummaryList: obj.InstanceSummaryList.map(function (item) { return InstanceSummary.filterSensitiveLog(item); }),
    }))); };
})(ListInstancesResponse || (ListInstancesResponse = {}));
export var ListInstanceStorageConfigsRequest;
(function (ListInstanceStorageConfigsRequest) {
    ListInstanceStorageConfigsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListInstanceStorageConfigsRequest || (ListInstanceStorageConfigsRequest = {}));
export var ListInstanceStorageConfigsResponse;
(function (ListInstanceStorageConfigsResponse) {
    ListInstanceStorageConfigsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListInstanceStorageConfigsResponse || (ListInstanceStorageConfigsResponse = {}));
export var ListIntegrationAssociationsRequest;
(function (ListIntegrationAssociationsRequest) {
    ListIntegrationAssociationsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListIntegrationAssociationsRequest || (ListIntegrationAssociationsRequest = {}));
export var IntegrationAssociationSummary;
(function (IntegrationAssociationSummary) {
    IntegrationAssociationSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(IntegrationAssociationSummary || (IntegrationAssociationSummary = {}));
export var ListIntegrationAssociationsResponse;
(function (ListIntegrationAssociationsResponse) {
    ListIntegrationAssociationsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListIntegrationAssociationsResponse || (ListIntegrationAssociationsResponse = {}));
export var ListLambdaFunctionsRequest;
(function (ListLambdaFunctionsRequest) {
    ListLambdaFunctionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListLambdaFunctionsRequest || (ListLambdaFunctionsRequest = {}));
export var ListLambdaFunctionsResponse;
(function (ListLambdaFunctionsResponse) {
    ListLambdaFunctionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListLambdaFunctionsResponse || (ListLambdaFunctionsResponse = {}));
export var ListLexBotsRequest;
(function (ListLexBotsRequest) {
    ListLexBotsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListLexBotsRequest || (ListLexBotsRequest = {}));
export var ListLexBotsResponse;
(function (ListLexBotsResponse) {
    ListLexBotsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListLexBotsResponse || (ListLexBotsResponse = {}));
export var PhoneNumberCountryCode;
(function (PhoneNumberCountryCode) {
    PhoneNumberCountryCode["AD"] = "AD";
    PhoneNumberCountryCode["AE"] = "AE";
    PhoneNumberCountryCode["AF"] = "AF";
    PhoneNumberCountryCode["AG"] = "AG";
    PhoneNumberCountryCode["AI"] = "AI";
    PhoneNumberCountryCode["AL"] = "AL";
    PhoneNumberCountryCode["AM"] = "AM";
    PhoneNumberCountryCode["AN"] = "AN";
    PhoneNumberCountryCode["AO"] = "AO";
    PhoneNumberCountryCode["AQ"] = "AQ";
    PhoneNumberCountryCode["AR"] = "AR";
    PhoneNumberCountryCode["AS"] = "AS";
    PhoneNumberCountryCode["AT"] = "AT";
    PhoneNumberCountryCode["AU"] = "AU";
    PhoneNumberCountryCode["AW"] = "AW";
    PhoneNumberCountryCode["AZ"] = "AZ";
    PhoneNumberCountryCode["BA"] = "BA";
    PhoneNumberCountryCode["BB"] = "BB";
    PhoneNumberCountryCode["BD"] = "BD";
    PhoneNumberCountryCode["BE"] = "BE";
    PhoneNumberCountryCode["BF"] = "BF";
    PhoneNumberCountryCode["BG"] = "BG";
    PhoneNumberCountryCode["BH"] = "BH";
    PhoneNumberCountryCode["BI"] = "BI";
    PhoneNumberCountryCode["BJ"] = "BJ";
    PhoneNumberCountryCode["BL"] = "BL";
    PhoneNumberCountryCode["BM"] = "BM";
    PhoneNumberCountryCode["BN"] = "BN";
    PhoneNumberCountryCode["BO"] = "BO";
    PhoneNumberCountryCode["BR"] = "BR";
    PhoneNumberCountryCode["BS"] = "BS";
    PhoneNumberCountryCode["BT"] = "BT";
    PhoneNumberCountryCode["BW"] = "BW";
    PhoneNumberCountryCode["BY"] = "BY";
    PhoneNumberCountryCode["BZ"] = "BZ";
    PhoneNumberCountryCode["CA"] = "CA";
    PhoneNumberCountryCode["CC"] = "CC";
    PhoneNumberCountryCode["CD"] = "CD";
    PhoneNumberCountryCode["CF"] = "CF";
    PhoneNumberCountryCode["CG"] = "CG";
    PhoneNumberCountryCode["CH"] = "CH";
    PhoneNumberCountryCode["CI"] = "CI";
    PhoneNumberCountryCode["CK"] = "CK";
    PhoneNumberCountryCode["CL"] = "CL";
    PhoneNumberCountryCode["CM"] = "CM";
    PhoneNumberCountryCode["CN"] = "CN";
    PhoneNumberCountryCode["CO"] = "CO";
    PhoneNumberCountryCode["CR"] = "CR";
    PhoneNumberCountryCode["CU"] = "CU";
    PhoneNumberCountryCode["CV"] = "CV";
    PhoneNumberCountryCode["CW"] = "CW";
    PhoneNumberCountryCode["CX"] = "CX";
    PhoneNumberCountryCode["CY"] = "CY";
    PhoneNumberCountryCode["CZ"] = "CZ";
    PhoneNumberCountryCode["DE"] = "DE";
    PhoneNumberCountryCode["DJ"] = "DJ";
    PhoneNumberCountryCode["DK"] = "DK";
    PhoneNumberCountryCode["DM"] = "DM";
    PhoneNumberCountryCode["DO"] = "DO";
    PhoneNumberCountryCode["DZ"] = "DZ";
    PhoneNumberCountryCode["EC"] = "EC";
    PhoneNumberCountryCode["EE"] = "EE";
    PhoneNumberCountryCode["EG"] = "EG";
    PhoneNumberCountryCode["EH"] = "EH";
    PhoneNumberCountryCode["ER"] = "ER";
    PhoneNumberCountryCode["ES"] = "ES";
    PhoneNumberCountryCode["ET"] = "ET";
    PhoneNumberCountryCode["FI"] = "FI";
    PhoneNumberCountryCode["FJ"] = "FJ";
    PhoneNumberCountryCode["FK"] = "FK";
    PhoneNumberCountryCode["FM"] = "FM";
    PhoneNumberCountryCode["FO"] = "FO";
    PhoneNumberCountryCode["FR"] = "FR";
    PhoneNumberCountryCode["GA"] = "GA";
    PhoneNumberCountryCode["GB"] = "GB";
    PhoneNumberCountryCode["GD"] = "GD";
    PhoneNumberCountryCode["GE"] = "GE";
    PhoneNumberCountryCode["GG"] = "GG";
    PhoneNumberCountryCode["GH"] = "GH";
    PhoneNumberCountryCode["GI"] = "GI";
    PhoneNumberCountryCode["GL"] = "GL";
    PhoneNumberCountryCode["GM"] = "GM";
    PhoneNumberCountryCode["GN"] = "GN";
    PhoneNumberCountryCode["GQ"] = "GQ";
    PhoneNumberCountryCode["GR"] = "GR";
    PhoneNumberCountryCode["GT"] = "GT";
    PhoneNumberCountryCode["GU"] = "GU";
    PhoneNumberCountryCode["GW"] = "GW";
    PhoneNumberCountryCode["GY"] = "GY";
    PhoneNumberCountryCode["HK"] = "HK";
    PhoneNumberCountryCode["HN"] = "HN";
    PhoneNumberCountryCode["HR"] = "HR";
    PhoneNumberCountryCode["HT"] = "HT";
    PhoneNumberCountryCode["HU"] = "HU";
    PhoneNumberCountryCode["ID"] = "ID";
    PhoneNumberCountryCode["IE"] = "IE";
    PhoneNumberCountryCode["IL"] = "IL";
    PhoneNumberCountryCode["IM"] = "IM";
    PhoneNumberCountryCode["IN"] = "IN";
    PhoneNumberCountryCode["IO"] = "IO";
    PhoneNumberCountryCode["IQ"] = "IQ";
    PhoneNumberCountryCode["IR"] = "IR";
    PhoneNumberCountryCode["IS"] = "IS";
    PhoneNumberCountryCode["IT"] = "IT";
    PhoneNumberCountryCode["JE"] = "JE";
    PhoneNumberCountryCode["JM"] = "JM";
    PhoneNumberCountryCode["JO"] = "JO";
    PhoneNumberCountryCode["JP"] = "JP";
    PhoneNumberCountryCode["KE"] = "KE";
    PhoneNumberCountryCode["KG"] = "KG";
    PhoneNumberCountryCode["KH"] = "KH";
    PhoneNumberCountryCode["KI"] = "KI";
    PhoneNumberCountryCode["KM"] = "KM";
    PhoneNumberCountryCode["KN"] = "KN";
    PhoneNumberCountryCode["KP"] = "KP";
    PhoneNumberCountryCode["KR"] = "KR";
    PhoneNumberCountryCode["KW"] = "KW";
    PhoneNumberCountryCode["KY"] = "KY";
    PhoneNumberCountryCode["KZ"] = "KZ";
    PhoneNumberCountryCode["LA"] = "LA";
    PhoneNumberCountryCode["LB"] = "LB";
    PhoneNumberCountryCode["LC"] = "LC";
    PhoneNumberCountryCode["LI"] = "LI";
    PhoneNumberCountryCode["LK"] = "LK";
    PhoneNumberCountryCode["LR"] = "LR";
    PhoneNumberCountryCode["LS"] = "LS";
    PhoneNumberCountryCode["LT"] = "LT";
    PhoneNumberCountryCode["LU"] = "LU";
    PhoneNumberCountryCode["LV"] = "LV";
    PhoneNumberCountryCode["LY"] = "LY";
    PhoneNumberCountryCode["MA"] = "MA";
    PhoneNumberCountryCode["MC"] = "MC";
    PhoneNumberCountryCode["MD"] = "MD";
    PhoneNumberCountryCode["ME"] = "ME";
    PhoneNumberCountryCode["MF"] = "MF";
    PhoneNumberCountryCode["MG"] = "MG";
    PhoneNumberCountryCode["MH"] = "MH";
    PhoneNumberCountryCode["MK"] = "MK";
    PhoneNumberCountryCode["ML"] = "ML";
    PhoneNumberCountryCode["MM"] = "MM";
    PhoneNumberCountryCode["MN"] = "MN";
    PhoneNumberCountryCode["MO"] = "MO";
    PhoneNumberCountryCode["MP"] = "MP";
    PhoneNumberCountryCode["MR"] = "MR";
    PhoneNumberCountryCode["MS"] = "MS";
    PhoneNumberCountryCode["MT"] = "MT";
    PhoneNumberCountryCode["MU"] = "MU";
    PhoneNumberCountryCode["MV"] = "MV";
    PhoneNumberCountryCode["MW"] = "MW";
    PhoneNumberCountryCode["MX"] = "MX";
    PhoneNumberCountryCode["MY"] = "MY";
    PhoneNumberCountryCode["MZ"] = "MZ";
    PhoneNumberCountryCode["NA"] = "NA";
    PhoneNumberCountryCode["NC"] = "NC";
    PhoneNumberCountryCode["NE"] = "NE";
    PhoneNumberCountryCode["NG"] = "NG";
    PhoneNumberCountryCode["NI"] = "NI";
    PhoneNumberCountryCode["NL"] = "NL";
    PhoneNumberCountryCode["NO"] = "NO";
    PhoneNumberCountryCode["NP"] = "NP";
    PhoneNumberCountryCode["NR"] = "NR";
    PhoneNumberCountryCode["NU"] = "NU";
    PhoneNumberCountryCode["NZ"] = "NZ";
    PhoneNumberCountryCode["OM"] = "OM";
    PhoneNumberCountryCode["PA"] = "PA";
    PhoneNumberCountryCode["PE"] = "PE";
    PhoneNumberCountryCode["PF"] = "PF";
    PhoneNumberCountryCode["PG"] = "PG";
    PhoneNumberCountryCode["PH"] = "PH";
    PhoneNumberCountryCode["PK"] = "PK";
    PhoneNumberCountryCode["PL"] = "PL";
    PhoneNumberCountryCode["PM"] = "PM";
    PhoneNumberCountryCode["PN"] = "PN";
    PhoneNumberCountryCode["PR"] = "PR";
    PhoneNumberCountryCode["PT"] = "PT";
    PhoneNumberCountryCode["PW"] = "PW";
    PhoneNumberCountryCode["PY"] = "PY";
    PhoneNumberCountryCode["QA"] = "QA";
    PhoneNumberCountryCode["RE"] = "RE";
    PhoneNumberCountryCode["RO"] = "RO";
    PhoneNumberCountryCode["RS"] = "RS";
    PhoneNumberCountryCode["RU"] = "RU";
    PhoneNumberCountryCode["RW"] = "RW";
    PhoneNumberCountryCode["SA"] = "SA";
    PhoneNumberCountryCode["SB"] = "SB";
    PhoneNumberCountryCode["SC"] = "SC";
    PhoneNumberCountryCode["SD"] = "SD";
    PhoneNumberCountryCode["SE"] = "SE";
    PhoneNumberCountryCode["SG"] = "SG";
    PhoneNumberCountryCode["SH"] = "SH";
    PhoneNumberCountryCode["SI"] = "SI";
    PhoneNumberCountryCode["SJ"] = "SJ";
    PhoneNumberCountryCode["SK"] = "SK";
    PhoneNumberCountryCode["SL"] = "SL";
    PhoneNumberCountryCode["SM"] = "SM";
    PhoneNumberCountryCode["SN"] = "SN";
    PhoneNumberCountryCode["SO"] = "SO";
    PhoneNumberCountryCode["SR"] = "SR";
    PhoneNumberCountryCode["ST"] = "ST";
    PhoneNumberCountryCode["SV"] = "SV";
    PhoneNumberCountryCode["SX"] = "SX";
    PhoneNumberCountryCode["SY"] = "SY";
    PhoneNumberCountryCode["SZ"] = "SZ";
    PhoneNumberCountryCode["TC"] = "TC";
    PhoneNumberCountryCode["TD"] = "TD";
    PhoneNumberCountryCode["TG"] = "TG";
    PhoneNumberCountryCode["TH"] = "TH";
    PhoneNumberCountryCode["TJ"] = "TJ";
    PhoneNumberCountryCode["TK"] = "TK";
    PhoneNumberCountryCode["TL"] = "TL";
    PhoneNumberCountryCode["TM"] = "TM";
    PhoneNumberCountryCode["TN"] = "TN";
    PhoneNumberCountryCode["TO"] = "TO";
    PhoneNumberCountryCode["TR"] = "TR";
    PhoneNumberCountryCode["TT"] = "TT";
    PhoneNumberCountryCode["TV"] = "TV";
    PhoneNumberCountryCode["TW"] = "TW";
    PhoneNumberCountryCode["TZ"] = "TZ";
    PhoneNumberCountryCode["UA"] = "UA";
    PhoneNumberCountryCode["UG"] = "UG";
    PhoneNumberCountryCode["US"] = "US";
    PhoneNumberCountryCode["UY"] = "UY";
    PhoneNumberCountryCode["UZ"] = "UZ";
    PhoneNumberCountryCode["VA"] = "VA";
    PhoneNumberCountryCode["VC"] = "VC";
    PhoneNumberCountryCode["VE"] = "VE";
    PhoneNumberCountryCode["VG"] = "VG";
    PhoneNumberCountryCode["VI"] = "VI";
    PhoneNumberCountryCode["VN"] = "VN";
    PhoneNumberCountryCode["VU"] = "VU";
    PhoneNumberCountryCode["WF"] = "WF";
    PhoneNumberCountryCode["WS"] = "WS";
    PhoneNumberCountryCode["YE"] = "YE";
    PhoneNumberCountryCode["YT"] = "YT";
    PhoneNumberCountryCode["ZA"] = "ZA";
    PhoneNumberCountryCode["ZM"] = "ZM";
    PhoneNumberCountryCode["ZW"] = "ZW";
})(PhoneNumberCountryCode || (PhoneNumberCountryCode = {}));
export var PhoneNumberType;
(function (PhoneNumberType) {
    PhoneNumberType["DID"] = "DID";
    PhoneNumberType["TOLL_FREE"] = "TOLL_FREE";
})(PhoneNumberType || (PhoneNumberType = {}));
export var ListPhoneNumbersRequest;
(function (ListPhoneNumbersRequest) {
    ListPhoneNumbersRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPhoneNumbersRequest || (ListPhoneNumbersRequest = {}));
export var PhoneNumberSummary;
(function (PhoneNumberSummary) {
    PhoneNumberSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PhoneNumberSummary || (PhoneNumberSummary = {}));
export var ListPhoneNumbersResponse;
(function (ListPhoneNumbersResponse) {
    ListPhoneNumbersResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPhoneNumbersResponse || (ListPhoneNumbersResponse = {}));
export var ListPromptsRequest;
(function (ListPromptsRequest) {
    ListPromptsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPromptsRequest || (ListPromptsRequest = {}));
export var PromptSummary;
(function (PromptSummary) {
    PromptSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PromptSummary || (PromptSummary = {}));
export var ListPromptsResponse;
(function (ListPromptsResponse) {
    ListPromptsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPromptsResponse || (ListPromptsResponse = {}));
export var ListQueueQuickConnectsRequest;
(function (ListQueueQuickConnectsRequest) {
    ListQueueQuickConnectsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListQueueQuickConnectsRequest || (ListQueueQuickConnectsRequest = {}));
export var QuickConnectSummary;
(function (QuickConnectSummary) {
    QuickConnectSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(QuickConnectSummary || (QuickConnectSummary = {}));
export var ListQueueQuickConnectsResponse;
(function (ListQueueQuickConnectsResponse) {
    ListQueueQuickConnectsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListQueueQuickConnectsResponse || (ListQueueQuickConnectsResponse = {}));
export var QueueType;
(function (QueueType) {
    QueueType["AGENT"] = "AGENT";
    QueueType["STANDARD"] = "STANDARD";
})(QueueType || (QueueType = {}));
export var ListQueuesRequest;
(function (ListQueuesRequest) {
    ListQueuesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListQueuesRequest || (ListQueuesRequest = {}));
export var QueueSummary;
(function (QueueSummary) {
    QueueSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(QueueSummary || (QueueSummary = {}));
export var ListQueuesResponse;
(function (ListQueuesResponse) {
    ListQueuesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListQueuesResponse || (ListQueuesResponse = {}));
export var ListQuickConnectsRequest;
(function (ListQuickConnectsRequest) {
    ListQuickConnectsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListQuickConnectsRequest || (ListQuickConnectsRequest = {}));
export var ListQuickConnectsResponse;
(function (ListQuickConnectsResponse) {
    ListQuickConnectsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListQuickConnectsResponse || (ListQuickConnectsResponse = {}));
export var ListRoutingProfileQueuesRequest;
(function (ListRoutingProfileQueuesRequest) {
    ListRoutingProfileQueuesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListRoutingProfileQueuesRequest || (ListRoutingProfileQueuesRequest = {}));
export var RoutingProfileQueueConfigSummary;
(function (RoutingProfileQueueConfigSummary) {
    RoutingProfileQueueConfigSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RoutingProfileQueueConfigSummary || (RoutingProfileQueueConfigSummary = {}));
export var ListRoutingProfileQueuesResponse;
(function (ListRoutingProfileQueuesResponse) {
    ListRoutingProfileQueuesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListRoutingProfileQueuesResponse || (ListRoutingProfileQueuesResponse = {}));
export var ListRoutingProfilesRequest;
(function (ListRoutingProfilesRequest) {
    ListRoutingProfilesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListRoutingProfilesRequest || (ListRoutingProfilesRequest = {}));
export var RoutingProfileSummary;
(function (RoutingProfileSummary) {
    RoutingProfileSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RoutingProfileSummary || (RoutingProfileSummary = {}));
export var ListRoutingProfilesResponse;
(function (ListRoutingProfilesResponse) {
    ListRoutingProfilesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListRoutingProfilesResponse || (ListRoutingProfilesResponse = {}));
export var ListSecurityKeysRequest;
(function (ListSecurityKeysRequest) {
    ListSecurityKeysRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSecurityKeysRequest || (ListSecurityKeysRequest = {}));
export var SecurityKey;
(function (SecurityKey) {
    SecurityKey.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SecurityKey || (SecurityKey = {}));
export var ListSecurityKeysResponse;
(function (ListSecurityKeysResponse) {
    ListSecurityKeysResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSecurityKeysResponse || (ListSecurityKeysResponse = {}));
export var ListSecurityProfilePermissionsRequest;
(function (ListSecurityProfilePermissionsRequest) {
    ListSecurityProfilePermissionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSecurityProfilePermissionsRequest || (ListSecurityProfilePermissionsRequest = {}));
export var ListSecurityProfilePermissionsResponse;
(function (ListSecurityProfilePermissionsResponse) {
    ListSecurityProfilePermissionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSecurityProfilePermissionsResponse || (ListSecurityProfilePermissionsResponse = {}));
export var ListSecurityProfilesRequest;
(function (ListSecurityProfilesRequest) {
    ListSecurityProfilesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSecurityProfilesRequest || (ListSecurityProfilesRequest = {}));
export var SecurityProfileSummary;
(function (SecurityProfileSummary) {
    SecurityProfileSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SecurityProfileSummary || (SecurityProfileSummary = {}));
export var ListSecurityProfilesResponse;
(function (ListSecurityProfilesResponse) {
    ListSecurityProfilesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSecurityProfilesResponse || (ListSecurityProfilesResponse = {}));
export var ListTagsForResourceRequest;
(function (ListTagsForResourceRequest) {
    ListTagsForResourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTagsForResourceRequest || (ListTagsForResourceRequest = {}));
export var ListTagsForResourceResponse;
(function (ListTagsForResourceResponse) {
    ListTagsForResourceResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTagsForResourceResponse || (ListTagsForResourceResponse = {}));
export var ListUseCasesRequest;
(function (ListUseCasesRequest) {
    ListUseCasesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListUseCasesRequest || (ListUseCasesRequest = {}));
export var UseCase;
(function (UseCase) {
    UseCase.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UseCase || (UseCase = {}));
export var ListUseCasesResponse;
(function (ListUseCasesResponse) {
    ListUseCasesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListUseCasesResponse || (ListUseCasesResponse = {}));
export var ListUserHierarchyGroupsRequest;
(function (ListUserHierarchyGroupsRequest) {
    ListUserHierarchyGroupsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListUserHierarchyGroupsRequest || (ListUserHierarchyGroupsRequest = {}));
export var ListUserHierarchyGroupsResponse;
(function (ListUserHierarchyGroupsResponse) {
    ListUserHierarchyGroupsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListUserHierarchyGroupsResponse || (ListUserHierarchyGroupsResponse = {}));
export var ListUsersRequest;
(function (ListUsersRequest) {
    ListUsersRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListUsersRequest || (ListUsersRequest = {}));
export var UserSummary;
(function (UserSummary) {
    UserSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UserSummary || (UserSummary = {}));
export var ListUsersResponse;
(function (ListUsersResponse) {
    ListUsersResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListUsersResponse || (ListUsersResponse = {}));
export var ResumeContactRecordingRequest;
(function (ResumeContactRecordingRequest) {
    ResumeContactRecordingRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResumeContactRecordingRequest || (ResumeContactRecordingRequest = {}));
export var ResumeContactRecordingResponse;
(function (ResumeContactRecordingResponse) {
    ResumeContactRecordingResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResumeContactRecordingResponse || (ResumeContactRecordingResponse = {}));
export var SearchVocabulariesRequest;
(function (SearchVocabulariesRequest) {
    SearchVocabulariesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SearchVocabulariesRequest || (SearchVocabulariesRequest = {}));
export var VocabularySummary;
(function (VocabularySummary) {
    VocabularySummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(VocabularySummary || (VocabularySummary = {}));
export var SearchVocabulariesResponse;
(function (SearchVocabulariesResponse) {
    SearchVocabulariesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SearchVocabulariesResponse || (SearchVocabulariesResponse = {}));
export var ChatMessage;
(function (ChatMessage) {
    ChatMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ChatMessage || (ChatMessage = {}));
export var ParticipantDetails;
(function (ParticipantDetails) {
    ParticipantDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ParticipantDetails || (ParticipantDetails = {}));
export var StartChatContactRequest;
(function (StartChatContactRequest) {
    StartChatContactRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartChatContactRequest || (StartChatContactRequest = {}));
export var StartChatContactResponse;
(function (StartChatContactResponse) {
    StartChatContactResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartChatContactResponse || (StartChatContactResponse = {}));
export var VoiceRecordingTrack;
(function (VoiceRecordingTrack) {
    VoiceRecordingTrack["ALL"] = "ALL";
    VoiceRecordingTrack["FROM_AGENT"] = "FROM_AGENT";
    VoiceRecordingTrack["TO_AGENT"] = "TO_AGENT";
})(VoiceRecordingTrack || (VoiceRecordingTrack = {}));
export var VoiceRecordingConfiguration;
(function (VoiceRecordingConfiguration) {
    VoiceRecordingConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(VoiceRecordingConfiguration || (VoiceRecordingConfiguration = {}));
export var StartContactRecordingRequest;
(function (StartContactRecordingRequest) {
    StartContactRecordingRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartContactRecordingRequest || (StartContactRecordingRequest = {}));
export var StartContactRecordingResponse;
(function (StartContactRecordingResponse) {
    StartContactRecordingResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartContactRecordingResponse || (StartContactRecordingResponse = {}));
export var ChatStreamingConfiguration;
(function (ChatStreamingConfiguration) {
    ChatStreamingConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ChatStreamingConfiguration || (ChatStreamingConfiguration = {}));
export var StartContactStreamingRequest;
(function (StartContactStreamingRequest) {
    StartContactStreamingRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartContactStreamingRequest || (StartContactStreamingRequest = {}));
export var StartContactStreamingResponse;
(function (StartContactStreamingResponse) {
    StartContactStreamingResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartContactStreamingResponse || (StartContactStreamingResponse = {}));
