// @aws-sdk/client-connect/dist-cjs/commands/SuspendContactRecordingCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SuspendContactRecordingCommand = void 0;
const middleware_serde_1 = require("@aws-sdk/middleware-serde");
const smithy_client_1 = require("@aws-sdk/smithy-client");
const models_1_1 = require("../models/models_1");
const Aws_restJson1_1 = require("../protocols/Aws_restJson1");
class SuspendContactRecordingCommand extends smithy_client_1.Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(middleware_serde_1.getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "ConnectClient";
        const commandName = "SuspendContactRecordingCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: models_1_1.SuspendContactRecordingRequest.filterSensitiveLog,
            outputFilterSensitiveLog: models_1_1.SuspendContactRecordingResponse.filterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return Aws_restJson1_1.serializeAws_restJson1SuspendContactRecordingCommand(input, context);
    }
    deserialize(output, context) {
        return Aws_restJson1_1.deserializeAws_restJson1SuspendContactRecordingCommand(output, context);
    }
}
exports.SuspendContactRecordingCommand = SuspendContactRecordingCommand;
